diff --git a/tryton/modules/account_asset/tryton.cfg b/tryton/modules/account_asset/tryton.cfg
index bf83a4f3e5..ff33224ca8 100644
--- a/tryton/modules/account_asset/tryton.cfg
+++ b/tryton/modules/account_asset/tryton.cfg
@@ -15,10 +15,8 @@ xml:
     invoice.xml
     account.xml
     minimal_chart_bg.xml
-    minimal_chart_ca.xml
     minimal_chart_de.xml
     minimal_chart_en.xml
-    minimal_chart_es.xml
     minimal_chart_fr.xml
     minimal_chart_nl.xml
     minimal_chart_pt.xml
