diff --git a/tryton/modules/account/message.xml b/tryton/modules/account/message.xml
index 50b8dc4eda..5a138764ec 100644
--- a/tryton/modules/account/message.xml
+++ b/tryton/modules/account/message.xml
@@ -178,6 +178,9 @@ You may need to cancel its move "%(move)s".</field>
         <record model="ir.message" id="msg_reconciliation_unbalanced">
             <field name="text">To reconcile lines, they must have the same debit "%(debit)s" and credit "%(credit)s".</field>
         </record>
+        <record model="ir.message" id="msg_reconciliation_write_off_missing">
+            <field name="text">To reconcile lines with "%(amount)s" write-off, you must select a write-off method.</field>
+        </record>

         <record model="ir.message" id="msg_line_debit_credit">
             <field name="text">You cannot set both debit and credit on the line.</field>
diff --git a/tryton/modules/account/move.py b/tryton/modules/account/move.py
index fcb6cf9814..ef725e9297 100644
--- a/tryton/modules/account/move.py
+++ b/tryton/modules/account/move.py
@@ -1522,6 +1522,8 @@ class Line(MoveLineMixin, ModelSQL, ModelView):
         """
         pool = Pool()
         Reconciliation = pool.get('account.move.reconciliation')
+        Lang = pool.get('ir.lang')
+        lang = Lang.get()
         delegate_to = delegate_to.id if delegate_to else None

         reconciliations = []
@@ -1563,6 +1565,11 @@ class Line(MoveLineMixin, ModelSQL, ModelView):
                 writeoff_amount = amount
                 writeoff_currency = company.currency
             if writeoff_amount:
+                if not writeoff:
+                    raise ReconciliationError(gettext(
+                            'account.msg_reconciliation_write_off_missing',
+                            amount=lang.currency(
+                                writeoff_amount, writeoff_currency)))
                 move = cls._get_writeoff_move(
                     reconcile_account, reconcile_party,
                     writeoff_amount, writeoff_currency,
