diff --git a/modules/account_stock_eu/stock.py b/modules/account_stock_eu/stock.py
index 782bc1ec44..8605514aa4 100644
--- a/tryton/modules/account_stock_eu/stock.py
+++ b/tryton/modules/account_stock_eu/stock.py
@@ -571,6 +571,7 @@ class ShipmentIn(ShipmentMixin, metaclass=PoolMeta):
 
     @fields.depends('supplier')
     def on_change_supplier(self):
+        super().on_change_supplier()
         if self.supplier:
             address = self.supplier.address_get(type='delivery')
             if address:
