diff --git a/asset.py b/asset.py
index 371955c..dad6d8f 100644
--- a/trytond/trytond/modules/account_asset/asset.py	Mon May 06 14:53:17 2019 +0200
+++ b/trytond/trytond/modules/account_asset/asset.py	Tue Jun 11 22:56:45 2019 +0200
@@ -711,7 +711,7 @@ class UpdateAssetShowDepreciation(ModelView):
     date = fields.Date('Date', required=True,
         domain=[
             ('date', '>', Eval('latest_move_date')),
-            ('date', '<', Eval('next_depreciation_date')),
+            ('date', '<=', Eval('next_depreciation_date')),
             ],
         depends=['latest_move_date', 'next_depreciation_date'],
         help=('The date must be between the last update/depreciation date '
@@ -776,6 +776,9 @@ class UpdateAsset(Wizard):
     def default_show_move(self, fields):
         Asset = Pool().get('account.asset')
         asset = Asset(Transaction().context['active_id'])
+        if not isinstance(self.start.value, Decimal):
+            self.start.value = Decimal(self.start.value).quantize(
+                Decimal(str(10.0 ** -asset.currency_digits)))
         return {
             'amount': self.start.value - asset.value,
             'date': datetime.date.today(),
@@ -924,6 +927,8 @@ class AssetDepreciationTable(CompanyReport):
                 if (self.start_date < self.asset.start_date
                         or not self.asset_lines):
                     return 0
+                if not self.asset_lines:
+                    return 0
                 value = self.asset_lines[0].acquired_value
                 date = self.asset_lines[0].date
                 for line in self.update_lines:
