diff --git a/product.py b/product.py
index edb0f67..dcb0197 100644
--- a/tryton/modules/sale_price_list/product.py
+++ b/tryton/modules/sale_price_list/product.py
@@ -37,7 +37,7 @@ class Product(metaclass=PoolMeta):
                 uom = self.sale_uom
             unit_price = price_list.compute(
                  customer, self, unit_price, quantity, uom)
-            if price_list.tax_included and taxes:
+            if price_list.tax_included and taxes and unit_price is not None:
                 unit_price = Tax.reverse_compute(unit_price, taxes)
         return unit_price
 
