diff --git a/tryton/sao/src/window.js b/tryton/sao/src/window.js
index 2d9eca2af9..08a2fe20fb 100644
--- a/tryton/sao/src/window.js
+++ b/tryton/sao/src/window.js
@@ -882,7 +882,7 @@
                         'class': 'form-control input-sm',
                         'aria-labelledby': label.attr('id'),
                     }).val(date || '').appendTo(cell);
-                    input.css('width', date.length + 'ch');
+                    input.css('width', (date || '').length + 'ch');
                     input.uniqueId();
                 });
             });
