diff --git a/tryton/modules/account/account.py b/tryton/modules/account/account.py
index 114d960362..9d0fadc92b 100644
--- a/tryton/modules/account/account.py
+++ b/tryton/modules/account/account.py
@@ -2467,6 +2467,9 @@ class GeneralLedgerLine(DescriptionOriginMixin, ModelSQL, ModelView):
                 column = Column(move, 'description').as_(fname)
             elif fname == 'party_required':
                 column = Column(account, 'party_required').as_(fname)
+            elif fname in {'origin', 'description'}:
+                column = Coalesce(
+                    Column(line, fname), Column(move, fname)).as_(fname)
             elif (not field_line
                     or fname == 'state'
                     or isinstance(field_line, fields.Function)):
@@ -2508,8 +2511,10 @@ class GeneralLedgerLine(DescriptionOriginMixin, ModelSQL, ModelView):
 
     @classmethod
     def get_origin(cls):
-        Line = Pool().get('account.move.line')
-        return Line.get_origin()
+        pool = Pool()
+        Move = pool.get('account.move')
+        Line = pool.get('account.move.line')
+        return list(set(Move.get_origin()) | set(Line.get_origin()))
 
     def get_balance(self, name):
         pool = Pool()
