diff --git a/tryton/modules/purchase/purchase.py b/tryton/modules/purchase/purchase.py
index 2b08db6758..f3756e913d 100644
--- a/tryton/modules/purchase/purchase.py
+++ b/tryton/modules/purchase/purchase.py
@@ -1395,7 +1395,7 @@ class Line(sequence_ordered(), ModelSQL, ModelView):
         for invoice_line in self.invoice_lines:
             if invoice_line.type != 'line':
                 continue
-            if (invoice_line.invoice.state == 'paid'
+            if (invoice_line.invoice and invoice_line.invoice.state == 'paid'
                     or invoice_line.invoice in skips):
                 quantity -= UoM.compute_qty(
                     invoice_line.unit or self.unit, invoice_line.quantity,
