diff --git a/tryton/trytond/trytond/protocols/wrappers.py b/tryton/trytond/trytond/protocols/wrappers.py
index 6937c078df..b5f3d500e1 100644
--- a/tryton/trytond/trytond/protocols/wrappers.py
+++ b/tryton/trytond/trytond/protocols/wrappers.py
@@ -317,6 +317,7 @@ def wrapper(request, database_name, *args, **kwargs):
             if request.rpc_method:
                 raise
             else:
+                logger.exception("Exception when processing %s", request)
                 abort(HTTPStatus.INTERNAL_SERVER_ERROR, e)
         return result
     return wrapper
