diff --git a/tryton/modules/account/account.py b/tryton/modules/account/account.py
index 466460acce..0c24c963d2 100644
--- a/tryton/modules/account/account.py
+++ b/tryton/modules/account/account.py
@@ -1071,7 +1071,6 @@ def get_credit_debit(cls, accounts, names):
             col_type = int if name == 'line_count' else Decimal
             result[name] = defaultdict(col_type)
 
-        table = cls.__table__()
         line = MoveLine.__table__()
 
         for company, c_accounts in groupby(accounts, key=lambda a: a.company):
@@ -1079,7 +1078,7 @@ def get_credit_debit(cls, accounts, names):
             ids = [a.id for a in c_accounts]
             with transaction.set_context(company=company.id):
                 line_query, fiscalyear_ids = MoveLine.query_get(line)
-            columns = [table.id]
+            columns = [line.account]
             types = [None]
             for name in names:
                 if name == 'line_count':
@@ -1089,12 +1088,10 @@ def get_credit_debit(cls, accounts, names):
                     columns.append(
                         Sum(Coalesce(Column(line, name), 0)).as_(name))
                     types.append('NUMERIC')
-            where = fields.SQL_OPERATORS['in'](table.id, ids)
-            query = (table.join(line, 'LEFT',
-                    condition=line.account == table.id
-                    ).select(*columns,
+            where = fields.SQL_OPERATORS['in'](line.account, ids)
+            query = (line.select(*columns,
                     where=where & line_query,
-                    group_by=table.id))
+                    group_by=line.account))
             if backend.name == 'sqlite':
                 sqlite_apply_types(query, types)
             cursor.execute(*query)
