diff --git a/tryton/modules/sale/sale.py b/tryton/modules/sale/sale.py
index 433dbab435..71ba382e94 100644
--- a/tryton/modules/sale/sale.py
+++ b/tryton/modules/sale/sale.py
@@ -1044,7 +1044,7 @@ def _get_shipment_sale(self, Shipment, key):
         shipment = Shipment(**values)
         shipment.on_change_warehouse()
         if Shipment.__name__ == 'stock.shipment.out':
-            if self.shipment_address == self.warehouse.address:
+            if self.warehouse and self.shipment_address == self.warehouse.address:
                 shipment.delivery_address = shipment.warehouse.address
             else:
                 shipment.delivery_address = self.shipment_address
