diff --git a/tryton/trytond/trytond/backend/postgresql/table.py b/tryton/trytond/trytond/backend/postgresql/table.py
index 85c4ff440f..407eab3b84 100644
--- a/tryton/trytond/trytond/backend/postgresql/table.py
+++ b/tryton/trytond/trytond/backend/postgresql/table.py
@@ -55,11 +55,6 @@ def _init(self, model, history=False):
         if view_exists:
             return
 
-        if model.__doc__ and self.is_owner:
-            cursor.execute(SQL('COMMENT ON TABLE {} IS %s').format(
-                        Identifier(self.table_name)),
-                (model.__doc__,))
-
         def migrate_to_identity(table, column):
             previous_seq_name = f"{table}_{column}_seq"
             cursor.execute(
@@ -328,13 +323,6 @@ def add_column(self, column_name, sql_type, default=None, comment=''):
         match = VARCHAR_SIZE_RE.match(sql_type)
         field_size = int(match.group(1)) if match else None
 
-        def add_comment():
-            if comment and self.is_owner:
-                cursor.execute(
-                    SQL('COMMENT ON COLUMN {}.{} IS %s').format(
-                        Identifier(self.table_name),
-                        Identifier(column_name)),
-                    (comment,))
         if self.column_exist(column_name):
             if (column_name in ('create_date', 'write_date')
                     and column_type[1].lower() != 'timestamp(6)'):
@@ -346,7 +334,6 @@ def add_comment():
                         Identifier(self.table_name),
                         Identifier(column_name)))
 
-            add_comment()
             base_type = column_type[0].lower()
             typname = self._columns[column_name]['typname']
             if base_type != typname:
@@ -407,7 +394,6 @@ def add_comment():
                 Identifier(self.table_name),
                 Identifier(column_name),
                 SQL(column_type)))
-        add_comment()
 
         if default:
             # check if table is non-empty:
