diff --git a/tryton/modules/sale_credit_limit/party.py b/tryton/modules/sale_credit_limit/party.py
index ea452e878e..1fce328541 100644
--- a/tryton/modules/sale_credit_limit/party.py
+++ b/tryton/modules/sale_credit_limit/party.py
@@ -40,9 +40,12 @@ class Party(metaclass=PoolMeta):
                         continue
                     invoice = invoice_line.invoice
                     if invoice and invoice.move:
-                        quantity -= Uom.compute_qty(
-                            invoice_line.unit, invoice_line.quantity,
-                            line.unit, round=False)
+                        if invoice_line.unit and line.unit:
+                            quantity -= Uom.compute_qty(
+                                invoice_line.unit, invoice_line.quantity,
+                                line.unit, round=False)
+                        else:
+                            quantity -= invoice_line.quantity
                 if quantity > 0:
                     amount += Currency.compute(
                         sale.currency,
