diff --git a/tryton/modules/production_work/work.py b/tryton/modules/production_work/work.py
index 8a38666398..ea81b331e7 100644
--- a/tryton/modules/production_work/work.py
+++ b/tryton/modules/production_work/work.py
@@ -421,7 +421,7 @@ class WorkCycle(Workflow, ModelSQL, ModelView):
 
     def set_duration(self, now):
         if self.duration is None:
-            self.duration = now - self.write_date
+            self.duration = now - (self.write_date or self.create_date)
 
     def set_cost(self):
         if self.cost is None:
