diff --git a/tryton/modules/account_payment_clearing/payment.py b/tryton/modules/account_payment_clearing/payment.py
index 72c69df198..aacd386c9a 100644
--- a/tryton/modules/account_payment_clearing/payment.py
+++ b/tryton/modules/account_payment_clearing/payment.py
@@ -277,6 +277,8 @@ class Payment(metaclass=PoolMeta):
             counterpart.amount_second_currency = self.amount.copy_sign(
                 counterpart.debit - counterpart.credit)
             counterpart.second_currency = self.journal.currency
+        counterpart.party = (self.clearing_party
+            if counterpart.account and counterpart.account.party_required else None)
         move.lines = (line, counterpart)
         return move
 
