diff --git a/tryton/sao/src/screen.js b/tryton/sao/src/screen.js
index 934379cde5..20be2b0188 100644
--- a/tryton/sao/src/screen.js
+++ b/tryton/sao/src/screen.js
@@ -1597,7 +1597,6 @@
         },
         save_current: function() {
             var current_record = this.current_record;
-            let new_record = current_record.id < 0;
             if (!current_record) {
                 if (this.current_view &&
                     (this.current_view.view_type == 'tree') &&
@@ -1608,6 +1607,7 @@
                     return jQuery.when();
                 }
             }
+            let new_record = current_record.id < 0;
             if (this.current_view) {
                 this.current_view.set_value();
                 var fields = this.current_view.get_fields();
