diff --git a/tryton/sao/src/tab.js b/tryton/sao/src/tab.js
index faf3738c63..56b1f6f547 100644
--- a/tryton/sao/src/tab.js
+++ b/tryton/sao/src/tab.js
@@ -1203,7 +1203,7 @@
                 this.refresh_resources(true);
             });
             for (const file of files) {
-                Sao.common.get_file_data(file, window_.add_data);
+                Sao.common.get_file_data(file, window_.add_data.bind(window_));
             }
             jQuery.when.apply(jQuery, uris).then(function() {
                 function empty(value) {
