diff --git a/tryton/modules/account/move.py b/tryton/modules/account/move.py
--- a/tryton/modules/account/move.py
+++ b/tryton/modules/account/move.py
@@ -1573,8 +1573,9 @@
                     ('company', '=', company),
                     ('journal', '=', journal),
                     ])
+            toolbar = dict(toolbar)
+            action = list(toolbar['action'])
             for template in templates:
-                action = toolbar['action']
                 # Use template id for action id to auto-select the template
                 action.append({
                         'name': template.name,
@@ -1582,6 +1583,7 @@
                         'wiz_name': 'account.move.template.create',
                         'id': template.id,
                         })
+            toolbar['action'] = action
         return toolbar
 
     @classmethod
