diff --git a/tryton/modules/account/tax.py b/tryton/modules/account/tax.py
index 4d671b052d..179de4c6b0 100644
--- a/tryton/modules/account/tax.py
+++ b/tryton/modules/account/tax.py
@@ -1281,6 +1281,8 @@ class TaxableMixin(object):
         pool = Pool()
         Tax = pool.get('account.tax')
         Configuration = pool.get('account.configuration')
+        Date = pool.get('ir.date')
+        today = Date.today()
         context = Transaction().context
 
         all_taxes = {}
@@ -1300,7 +1302,7 @@ class TaxableMixin(object):
                     assert all(t.company == self.company for t in line.taxes)
                     l_taxes = Tax.compute(
                         Tax.browse(line.taxes), line.unit_price,
-                        line.quantity, line.tax_date or self.tax_date)
+                        line.quantity, line.tax_date or self.tax_date or today)
                     current_taxes = {}
                     for tax in l_taxes:
                         taxline = self._compute_tax_line(**tax)
