diff --git a/tryton/modules/account_payment_sepa/payment.py b/tryton/modules/account_payment_sepa/payment.py
index ca417a1636..21d7604676 100644
--- a/tryton/modules/account_payment_sepa/payment.py
+++ b/tryton/modules/account_payment_sepa/payment.py
@@ -239,7 +239,8 @@ class Group(metaclass=PoolMeta):
                 raise ProcessError(
                     gettext('account_payment_sepa'
                         '.msg_payment_process_no_bank',
-                        payment=payment.rec_name))
+                        payment=payment.rec_name,
+                        bank_account=payment.sepa_bank_account_number.account))
         to_write = []
         for key, payments in self.sepa_payments:
             to_write.append(payments)
