diff --git a/tryton/trytond/trytond/__init__.py b/tryton/trytond/trytond/__init__.py
index 19a2293af6..bca1f168dd 100644
--- a/tryton/trytond/trytond/__init__.py
+++ b/tryton/trytond/trytond/__init__.py
@@ -21,7 +21,7 @@ __series__ = '.'.join(__version__.split('.')[:2])
 os.environ.setdefault(
     'TRYTOND_APPNAME',
     os.path.basename(getattr(__main__, '__file__', 'trytond')))
-os.environ['TRYTOND_TZ'] = os.environ.get('TZ') or 'UTC'
+os.environ.setdefault('TRYTOND_TZ', os.environ.get('TZ', 'UTC'))
 os.environ['TZ'] = 'UTC'
 if hasattr(time, 'tzset'):
     time.tzset()
