diff --git a/tryton/modules/account/journal.py b/tryton/modules/account/journal.py
index 27a2cc8c0d..271831a8ad 100644
--- a/tryton/modules/account/journal.py
+++ b/tryton/modules/account/journal.py
@@ -94,7 +94,7 @@ class Journal(
         pool = Pool()
         Company = pool.get('company.company')
         company_id = Transaction().context.get('company')
-        if company_id:
+        if company_id is not None and company_id >= 0:
             company = Company(company_id)
             currency_id = company.currency.id
         else:
diff --git a/tryton/modules/account/tax.py b/tryton/modules/account/tax.py
index 149d70e0bd..4d671b052d 100644
--- a/tryton/modules/account/tax.py
+++ b/tryton/modules/account/tax.py
@@ -1926,7 +1926,7 @@ class TestTaxView(ModelView, TaxableMixin):
         pool = Pool()
         Company = pool.get('company.company')
         company_id = Transaction().context.get('company')
-        if company_id:
+        if company_id is not None and company_id >= 0:
             company = Company(company_id)
             return company.currency.id

@@ -1935,7 +1935,7 @@ class TestTaxView(ModelView, TaxableMixin):
         pool = Pool()
         Company = pool.get('company.company')
         company_id = Transaction().context.get('company')
-        if company_id:
+        if company_id is not None and company_id >= 0:
             return Company(company_id)

     @property
diff --git a/tryton/modules/account_consolidation/account.py b/tryton/modules/account_consolidation/account.py
index 89e62b3815..dc47bcc1e0 100644
--- a/tryton/modules/account_consolidation/account.py
+++ b/tryton/modules/account_consolidation/account.py
@@ -322,7 +322,7 @@ class ConsolidationBalanceSheetContext(ModelView):
         pool = Pool()
         Company = pool.get('company.company')
         company_id = Transaction().context.get('company')
-        if company_id:
+        if company_id is not None and company_id >= 0:
             return Company(company_id).currency.id

     @classmethod
@@ -406,7 +406,7 @@ class ConsolidationIncomeStatementContext(ModelView):
         pool = Pool()
         Company = pool.get('company.company')
         company_id = Transaction().context.get('company')
-        if company_id:
+        if company_id is not None and company_id >= 0:
             return Company(company_id).currency.id

     @classmethod
diff --git a/tryton/modules/account_credit_limit/account.py b/tryton/modules/account_credit_limit/account.py
index 6157d4f90d..87801e4155 100644
--- a/tryton/modules/account_credit_limit/account.py
+++ b/tryton/modules/account_credit_limit/account.py
@@ -25,7 +25,7 @@ class Configuration(metaclass=PoolMeta):
         pool = Pool()
         Company = pool.get('company.company')
         company_id = Transaction().context.get('company')
-        if company_id:
+        if company_id is not None and company_id >= 0:
             company = Company(company_id)
             return company.currency.digits

diff --git a/tryton/modules/account_invoice/payment_term.py b/tryton/modules/account_invoice/payment_term.py
index 73c4bb607d..6dcd81327a 100644
--- a/tryton/modules/account_invoice/payment_term.py
+++ b/tryton/modules/account_invoice/payment_term.py
@@ -346,7 +346,7 @@ class TestPaymentTermView(ModelView):
         pool = Pool()
         Company = pool.get('company.company')
         company = Transaction().context.get('company')
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id

     @fields.depends('payment_term', 'date', 'amount', 'currency', 'result')
diff --git a/tryton/modules/account_payment_sepa/payment.py b/tryton/modules/account_payment_sepa/payment.py
index ae8e69d20d..ce5b2a8822 100644
--- a/tryton/modules/account_payment_sepa/payment.py
+++ b/tryton/modules/account_payment_sepa/payment.py
@@ -160,7 +160,7 @@ class Journal(metaclass=PoolMeta):
         pool = Pool()
         Company = pool.get('company.company')
         company_id = cls.default_company()
-        if company_id:
+        if company_id is not None and company_id >= 0:
             return Company(company_id).party.id

     @fields.depends('company')
diff --git a/tryton/modules/commission/commission.py b/tryton/modules/commission/commission.py
index 46ffc9f73f..11d04e6863 100644
--- a/tryton/modules/commission/commission.py
+++ b/tryton/modules/commission/commission.py
@@ -85,7 +85,7 @@ class Agent(ModelSQL, ModelView):
         pool = Pool()
         Company = pool.get('company.company')
         company = cls.default_company()
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id

     @fields.depends('company', 'currency')
diff --git a/tryton/modules/product_cost_history/product.py b/tryton/modules/product_cost_history/product.py
index 0755025bcc..974b660b95 100644
--- a/tryton/modules/product_cost_history/product.py
+++ b/tryton/modules/product_cost_history/product.py
@@ -25,7 +25,7 @@ class Product(metaclass=PoolMeta):
             datetime = context['_datetime']
             company = pattern.get(
                 'company', Transaction().context.get('company'))
-            if company:
+            if company is not None and company >= 0:
                 company = Company(company)
                 if company.timezone:
                     timezone = tz.ZoneInfo(company.timezone)
diff --git a/tryton/modules/product_cost_warehouse/product.py b/tryton/modules/product_cost_warehouse/product.py
index b58a637b77..5843fb620e 100644
--- a/tryton/modules/product_cost_warehouse/product.py
+++ b/tryton/modules/product_cost_warehouse/product.py
@@ -221,7 +221,7 @@ class CostPriceRevision(metaclass=PoolMeta):
         pool = Pool()
         Company = pool.get('company.company')
         company_id = cls.default_company()
-        if company_id:
+        if company_id is not None and company_id >= 0:
             company = Company(company_id)
             if company.cost_price_warehouse:
                 return Transaction().context.get('warehouse')
diff --git a/tryton/modules/purchase/purchase.py b/tryton/modules/purchase/purchase.py
index 2422879e2a..1ff77b3931 100644
--- a/tryton/modules/purchase/purchase.py
+++ b/tryton/modules/purchase/purchase.py
@@ -392,7 +392,7 @@ class Purchase(
         company = pattern.get('company')
         if not company:
             company = cls.default_company()
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id

     @classmethod
diff --git a/tryton/modules/purchase_blanket_agreement/purchase.py b/tryton/modules/purchase_blanket_agreement/purchase.py
index b616c9565d..d17d45b5c9 100644
--- a/tryton/modules/purchase_blanket_agreement/purchase.py
+++ b/tryton/modules/purchase_blanket_agreement/purchase.py
@@ -262,7 +262,7 @@ class BlanketAgreement(Workflow, ModelSQL, ModelView):
         company = pattern.get('company')
         if not company:
             company = cls.default_company()
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id

     @fields.depends('company', 'supplier', 'lines')
diff --git a/tryton/modules/purchase_requisition/purchase.py b/tryton/modules/purchase_requisition/purchase.py
index e7d220dbd9..5336372ffb 100644
--- a/tryton/modules/purchase_requisition/purchase.py
+++ b/tryton/modules/purchase_requisition/purchase.py
@@ -232,7 +232,7 @@ class PurchaseRequisition(Workflow, ModelSQL, ModelView):
     def default_currency(cls):
         Company = Pool().get('company.company')
         company = Transaction().context.get('company')
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id

     @fields.depends('lines', 'currency')
diff --git a/tryton/modules/sale/product.py b/tryton/modules/sale/product.py
index 9554687956..a9dc0158fd 100644
--- a/tryton/modules/sale/product.py
+++ b/tryton/modules/sale/product.py
@@ -260,7 +260,7 @@ class SaleContext(ModelView):
         pool = Pool()
         Company = pool.get('company.company')
         company_id = cls.default_company()
-        if company_id:
+        if company_id is not None and company_id >= 0:
             company = Company(company_id)
             return company.currency.id

diff --git a/tryton/modules/sale/sale.py b/tryton/modules/sale/sale.py
index 979a309070..7fa608de8a 100644
--- a/tryton/modules/sale/sale.py
+++ b/tryton/modules/sale/sale.py
@@ -450,7 +450,7 @@ class Sale(
         company = pattern.get('company')
         if not company:
             company = cls.default_company()
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id

     @classmethod
diff --git a/tryton/modules/sale/sale_reporting.py b/tryton/modules/sale/sale_reporting.py
index bdff1630db..99a81f5cd0 100644
--- a/tryton/modules/sale/sale_reporting.py
+++ b/tryton/modules/sale/sale_reporting.py
@@ -554,7 +554,7 @@ class CustomerCategoryTree(ModelSQL, ModelView):
         pool = Pool()
         Company = pool.get('company.company')
         company = Transaction().context.get('company')
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id


@@ -801,7 +801,7 @@ class ProductCategoryTree(ModelSQL, ModelView):
         pool = Pool()
         Company = pool.get('company.company')
         company = Transaction().context.get('company')
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id


@@ -928,7 +928,7 @@ class RegionTree(ModelSQL, ModelView):
         pool = Pool()
         Company = pool.get('company.company')
         company = Transaction().context.get('company')
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id


diff --git a/tryton/modules/sale_blanket_agreement/sale.py b/tryton/modules/sale_blanket_agreement/sale.py
index 1f2f8477d9..36fb215994 100644
--- a/tryton/modules/sale_blanket_agreement/sale.py
+++ b/tryton/modules/sale_blanket_agreement/sale.py
@@ -263,7 +263,7 @@ class BlanketAgreement(Workflow, ModelSQL, ModelView):
         company = pattern.get('company')
         if not company:
             company = cls.default_company()
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id

     @fields.depends('company', 'customer', 'lines')
diff --git a/tryton/modules/sale_opportunity/opportunity.py b/tryton/modules/sale_opportunity/opportunity.py
index 43ec0abf74..4aac7bccf9 100644
--- a/tryton/modules/sale_opportunity/opportunity.py
+++ b/tryton/modules/sale_opportunity/opportunity.py
@@ -256,7 +256,7 @@ class SaleOpportunity(
         company = pattern.get('company')
         if not company:
             company = cls.default_company()
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id

     @staticmethod
diff --git a/tryton/modules/sale_point/sale_reporting.py b/tryton/modules/sale_point/sale_reporting.py
index 6ea721a753..7c941cc5ae 100644
--- a/tryton/modules/sale_point/sale_reporting.py
+++ b/tryton/modules/sale_point/sale_reporting.py
@@ -20,7 +20,7 @@ class AbstractMixin:
         line = Line.__table__()
         point = Point.__table__()
         sale = Sale.__table__()
-        if company_id is not None:
+        if company_id is not None and company_id >= 0:
             company = Company(company_id)
             currency_id = company.currency.id
         else:
diff --git a/tryton/modules/sale_subscription/subscription.py b/tryton/modules/sale_subscription/subscription.py
index d5c02676b8..99d30a052c 100644
--- a/tryton/modules/sale_subscription/subscription.py
+++ b/tryton/modules/sale_subscription/subscription.py
@@ -241,7 +241,7 @@ class Subscription(Workflow, ModelSQL, ModelView):
         company = pattern.get('company')
         if not company:
             company = cls.default_company()
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id

     @classmethod
diff --git a/tryton/modules/stock/stock_reporting_margin.py b/tryton/modules/stock/stock_reporting_margin.py
index b3bc1101d4..658cb00532 100644
--- a/tryton/modules/stock/stock_reporting_margin.py
+++ b/tryton/modules/stock/stock_reporting_margin.py
@@ -637,7 +637,7 @@ class CategoryTree(ModelSQL, ModelView):
         pool = Pool()
         Company = pool.get('company.company')
         company = Transaction().context.get('company')
-        if company:
+        if company is not None and company >= 0:
             return Company(company).currency.id

     @classmethod
diff --git a/tryton/modules/web_shop/web.py b/tryton/modules/web_shop/web.py
index c16076be3b..aa99ebf9f0 100644
--- a/tryton/modules/web_shop/web.py
+++ b/tryton/modules/web_shop/web.py
@@ -133,7 +133,7 @@ class Shop(DeactivableMixin, ModelSQL, ModelView):
         pool = Pool()
         Company = pool.get('company.company')
         company_id = cls.default_company()
-        if company_id:
+        if company_id is not None and company_id >= 0:
             company = Company(company_id)
             return company.currency.id
