diff --git a/stock.py b/stock.py
index 6d831ce..e13a83f 100644
--- a/trytond/trytond/modules/account_stock_continental/stock.py
+++ b/trytond/trytond/modules/account_stock_continental/stock.py
@@ -102,11 +102,22 @@ class Move(metaclass=PoolMeta):
         if self.product.type != 'goods':
             return
 
-        date = self.effective_date or Date.today()
-        period_id = Period.find(self.company.id, date=date)
+        today = Date.today()
+        for date in [self.effective_date, today]:
+            period_id = Period.find(
+                self.company.id, date=date, exception=False, test_state=False)
+            if not period_id and date < today:
+                return
+            elif period_id:
+                break
+        else:
+            return
         period = Period(period_id)
         if not period.fiscalyear.account_stock_method:
             return
+        if period.state != 'open':
+            date = today
+            period = Period(Period.find(self.company.id, date=date))
 
         type_ = self._get_account_stock_move_type()
         if not type_:
