diff --git a/tryton/sao/src/session.js b/tryton/sao/src/session.js
index 8c1006d2ab..ae23a56fe4 100644
--- a/tryton/sao/src/session.js
+++ b/tryton/sao/src/session.js
@@ -113,7 +113,7 @@
             var prm = Sao.rpc(args, this);
             return prm.then(context => {
                 context = Object.fromEntries(Object.entries(context).filter(
-                    ([k, v]) => (k != 'locale') && !k.endsWith('.rec_name')));
+                    ([k, v]) => (k != 'locale')));
                 jQuery.extend(this.context, context);
                 this.store_context();
             });
