diff --git a/tryton/trytond/trytond/model/modelstorage.py b/tryton/trytond/trytond/model/modelstorage.py
index 7941d52b23..7d20a327fd 100644
--- a/tryton/trytond/trytond/model/modelstorage.py
+++ b/tryton/trytond/trytond/model/modelstorage.py
@@ -1437,10 +1437,11 @@ class ModelStorage(Model):
                             or any(d is None for d in digits)):
                         return
                     if isinstance(value, Decimal):
-                        if value.as_tuple().exponent < -digits[1]:
+                        exp = Decimal('.'.join(['0', '0' * digits[1]]))
+                        if value.quantize(exp) != value:
                             raise_error(value)
                     else:
-                        if round(value, digits[1]) != value:
+                        if not (round(value, digits[1]) == float(value)):
                             raise_error(value)
                 # validate digits
                 if getattr(field, 'digits', None):
