diff --git a/tryton/modules/account_payment/account.py b/tryton/modules/account_payment/account.py
index 112c8e1f2c..3a5b860aec 100644
--- a/tryton/modules/account_payment/account.py
+++ b/tryton/modules/account_payment/account.py
@@ -699,9 +699,9 @@ class Dunning(metaclass=PoolMeta):

     @classmethod
     def search_active(cls, name, clause):
-        if tuple(clause[1:]) in {('=', True), ('!=', False)}:
+        if tuple(clause[1:]) in [('=', True), ('!=', False)]:
             domain = ('line.payment_amount', '>', 0)
-        elif tuple(clause[1:]) in {('=', False), ('!=', True)}:
+        elif tuple(clause[1:]) in [('=', False), ('!=', True)]:
             domain = ('line.payment_amount', '<=', 0)
         else:
             domain = []
