diff --git a/tryton/modules/purchase/purchase.py b/tryton/modules/purchase/purchase.py
index 88a19980a9..195ff0e7d0 100644
--- a/tryton/modules/purchase/purchase.py
+++ b/tryton/modules/purchase/purchase.py
@@ -2030,7 +2030,7 @@ class Line(sequence_ordered(), ModelSQL, ModelView):
         quantity = (
             self._get_move_quantity(move_type)
             - self._get_shipped_quantity(move_type))
-        if quantity < 0:
+        if self.unit.round(quantity) < 0:
             warning_name = Warning.format(
                 'check_move_quantity', [self])
             if Warning.check(warning_name):
diff --git a/tryton/modules/sale/sale.py b/tryton/modules/sale/sale.py
index 4afac0e025..73d6bbb93c 100644
--- a/tryton/modules/sale/sale.py
+++ b/tryton/modules/sale/sale.py
@@ -2073,7 +2073,7 @@ class SaleLine(TaxableMixin, sequence_ordered(), ModelSQL, ModelView):
         quantity = (
             self._get_move_quantity(move_type)
             - self._get_shipped_quantity(move_type))
-        if quantity < 0:
+        if self.unit.round(quantity) < 0:
             warning_name = Warning.format(
                 'check_move_quantity', [self])
             if Warning.check(warning_name):
