diff --git a/tryton/trytond/trytond/pyson.py b/tryton/trytond/trytond/pyson.py
index 3ee9e6064d..ef398fc37e 100644
--- a/tryton/trytond/trytond/pyson.py
+++ b/tryton/trytond/trytond/pyson.py
@@ -149,7 +149,7 @@ class Eval(PYSON):
     @property
     def __repr_params__(self):
         params = (self._value,)
-        if self._default != '':
+        if isinstance(self._default, PYSON) or self._default != '':
             params += (self._default,)
         return params
