diff --git a/trytond/trytond/modules/stock/shipment.py b/trytond/trytond/modules/stock/shipment.py
index d176518..b915c21 100644
--- a/trytond/trytond/modules/stock/shipment.py
+++ b/trytond/trytond/modules/stock/shipment.py
@@ -1150,6 +1150,8 @@ class ShipmentOut(ShipmentMixin, Workflow, ModelSQL, ModelView):
         Move = pool.get('stock.move')
         Move.do([m for s in shipments for m in s.inventory_moves])
         cls._sync_inventory_to_outgoing(shipments, quantity=True)
+        Move.delete([m for s in shipments for m in s.outgoing_moves
+            if not m.quantity])
         Move.assign([m for s in shipments for m in s.outgoing_moves])

     def _sync_move_key(self, move):
