diff --git a/tryton/modules/project_revenue/work.py b/tryton/modules/project_revenue/work.py
index 644917f3eb..69b453a9df 100644
--- a/tryton/modules/project_revenue/work.py
+++ b/tryton/modules/project_revenue/work.py
@@ -172,32 +172,23 @@ class Work(metaclass=PoolMeta):
     def on_change_with_currency(self, name=None):
         return self.company.currency if self.company else None
 
-    @fields.depends('product', 'company')
+    @fields.depends('product')
     def on_change_product(self):
         pool = Pool()
-        User = pool.get('res.user')
         ModelData = pool.get('ir.model.data')
         Uom = pool.get('product.uom')
-        Currency = pool.get('currency.currency')

         if not self.product:
             return

-        if self.price_list_hour:
-            hour_uom = Uom(ModelData.get_id('product', 'uom_hour'))
-            self.list_price = Uom.compute_price(
-                self.product.default_uom, self.product.list_price, hour_uom)
-        else:
-            self.list_price = self.product.list_price
-
-        if self.company:
-            user = User(Transaction().user)
-            if user.company != self.company:
-                if user.company.currency != self.company.currency:
-                    self.list_price = Currency.compute(user.company.currency,
-                        self.list_price, self.company.currency, round=False)
-
-        self.list_price = round_price(self.list_price)
+        list_price = self.product.list_price_used
+        if list_price is not None:
+            if self.price_list_hour:
+                hour_uom = Uom(ModelData.get_id('product', 'uom_hour'))
+                list_price = Uom.compute_price(
+                    self.product.default_uom, list_price, hour_uom)
+            list_price = round_price(list_price)
+        self.list_price = list_price

     @property
     def price_list_hour(self):
