diff --git a/tryton/modules/commission/invoice.py b/tryton/modules/commission/invoice.py
index f7e2fb3d9e..b3dbee52ea 100644
--- a/tryton/modules/commission/invoice.py
+++ b/tryton/modules/commission/invoice.py
@@ -170,6 +170,7 @@ class InvoiceLine(metaclass=PoolMeta):
                 base_amount = Currency.compute(self.invoice.currency,
                     self.amount, agent.currency, round=False)
             amount = self._get_commission_amount(base_amount, plan)
+            base_amount = round_price(base_amount)
             if amount:
                 amount = round_price(amount)
             if not amount:
diff --git a/tryton/modules/commission/commission.py b/tryton/modules/commission/commission.py
index 08c693bdf1..2eabebc3e6 100644
--- a/tryton/modules/commission/commission.py
+++ b/tryton/modules/commission/commission.py
@@ -344,7 +344,7 @@ class Commission(ModelSQL, ModelView):
         states=_readonly_states,
         help="The product that is used on the invoice line.")
     base_amount = Monetary(
-        "Base Amount", currency='currency', digits='currency',
+        "Base Amount", currency='currency', digits=price_digits,
         states=_readonly_states)
     amount = Monetary(
         "Amount", currency='currency', required=True, digits=price_digits,
