diff --git a/tryton/sao/src/common.js b/tryton/sao/src/common.js
index c603e69436..e318701c22 100644
--- a/tryton/sao/src/common.js
+++ b/tryton/sao/src/common.js
@@ -3795,7 +3795,7 @@
         var sao_model = new Sao.Model(model);
         return sao_model.execute('search_read',
                 [domain, 0, Sao.config.limit, order, ['rec_name']], context).fail(function() {
-                    Sao.Logger.warning(
+                    Sao.Logger.warn(
                         "Unable to search for completion of %s", model);
                 });
     };
