diff --git a/tryton/modules/account_payment/payment.py b/tryton/modules/account_payment/payment.py
index 54cdb6c35f..a7fae0de18 100644
--- a/tryton/modules/account_payment/payment.py
+++ b/tryton/modules/account_payment/payment.py
@@ -295,6 +295,7 @@ class Payment(Workflow, ModelSQL, ModelView):
     date = fields.Date('Date', required=True, states=_STATES)
     amount = Monetary(
         "Amount", currency='currency', digits='currency', required=True,
+        domain=[('amount', '>', 0)],
         states={
             'readonly': ~Eval('state').in_(
                 If(Eval('process_method') == 'manual',
