diff --git a/tryton/sao/src/window.js b/tryton/sao/src/window.js
index 15e806b75f..8871e64a9b 100644
--- a/tryton/sao/src/window.js
+++ b/tryton/sao/src/window.js
@@ -416,7 +416,9 @@
                 }
             }
             let message = name + '/' + Sao.common.humanize(size);
-            this.label.text(message).attr('title', message);
+            if (this.label) {
+                this.label.text(message).attr('title', message);
+            }
             this._set_button_sensitive();
         },
         record_modified: function() {
