diff --git a/tryton/modules/purchase_request/purchase.py b/tryton/modules/purchase_request/purchase.py
index 63425351b4..a63b557763 100644
--- a/tryton/modules/purchase_request/purchase.py
+++ b/tryton/modules/purchase_request/purchase.py
@@ -26,6 +26,12 @@ def process_request(func):
 class Purchase(metaclass=PoolMeta):
     __name__ = 'purchase.purchase'
 
+    @classmethod
+    def copy(cls, lines, default=None):
+        default = default.copy() if default is not None else {}
+        default.setdefault('requests')
+        return super().copy(lines, default=default)
+
     @classmethod
     def delete(cls, purchases):
         cls.check_delete_purchase_request(purchases)
