diff --git a/tryton/modules/carrier_weight/carrier.py b/tryton/modules/carrier_weight/carrier.py
index 9ed0bec303..e310c8c6e5 100644
--- a/tryton/modules/carrier_weight/carrier.py
+++ b/tryton/modules/carrier_weight/carrier.py
@@ -46,6 +46,8 @@ class Carrier(metaclass=PoolMeta):

     def compute_weight_price(self, weight):
         "Compute price based on weight"
+        if not self.weight_price_list:
+            return Decimal(0)
         for line in reversed(self.weight_price_list):
             if line.weight < weight:
                 return line.price
