diff --git a/tryton/modules/project_invoice/project.py b/tryton/modules/project_invoice/project.py
index de31f26bb5..36ad3e4b0a 100644
--- a/tryton/modules/project_invoice/project.py
+++ b/tryton/modules/project_invoice/project.py
@@ -592,7 +592,7 @@ class Work(Effort, Progress, Timesheet, metaclass=PoolMeta):
         invoice_line.unit = key['unit']
         invoice_line.product = product
         invoice_line.on_change_product()
-        if not invoice_line.account:
+        if not getattr(invoice_line, 'account', None):
             if invoice_line.product:
                 raise InvoicingError(
                     gettext(
