diff --git a/tryton/modules/company/letter.fodt b/tryton/modules/company/letter.fodt
index 476ec8ab4c..74f063a645 100644
--- a/tryton/modules/company/letter.fodt
+++ b/tryton/modules/company/letter.fodt
@@ -349,15 +349,15 @@
  <office:master-styles>
   <style:master-page style:name="Standard" style:page-layout-name="pm1">
    <style:header>
-    <text:p text:style-name="P1"><text:placeholder text:placeholder-type="text">&lt;if test=&quot;user.company.header&quot;&gt;</text:placeholder></text:p>
+    <text:p text:style-name="P1"><text:placeholder text:placeholder-type="text">&lt;if test=&quot;user.company and user.company.header&quot;&gt;</text:placeholder></text:p>
     <text:p text:style-name="P1"><text:placeholder text:placeholder-type="text">&lt;for each=&quot;line in user.company.header.split(&apos;\n&apos;)&quot;&gt;</text:placeholder></text:p>
     <text:p text:style-name="P1"><text:placeholder text:placeholder-type="text">&lt;line&gt;</text:placeholder></text:p>
     <text:p text:style-name="P1"><text:placeholder text:placeholder-type="text">&lt;/for&gt;</text:placeholder></text:p>
     <text:p text:style-name="P2"><text:placeholder text:placeholder-type="text">&lt;/if&gt;</text:placeholder></text:p>
-    <text:p text:style-name="P3"><text:placeholder text:placeholder-type="text">&lt;user.company.rec_name&gt;</text:placeholder></text:p>
+    <text:p text:style-name="P3"><text:placeholder text:placeholder-type="text">&lt;user.company.rec_name if user.company else ''&gt;</text:placeholder></text:p>
    </style:header>
    <style:footer>
-    <text:p text:style-name="P2"><text:placeholder text:placeholder-type="text">&lt;if test=&quot;user.company.footer&quot;&gt;</text:placeholder></text:p>
+    <text:p text:style-name="P2"><text:placeholder text:placeholder-type="text">&lt;if test=&quot;user.company and user.company.footer&quot;&gt;</text:placeholder></text:p>
     <text:p text:style-name="P2"><text:placeholder text:placeholder-type="text">&lt;for each=&quot;line in user.company.footer.split(&apos;\n&apos;)&quot;&gt;</text:placeholder></text:p>
     <text:p text:style-name="P2"><text:placeholder text:placeholder-type="text">&lt;line&gt;</text:placeholder></text:p>
     <text:p text:style-name="P2"><text:placeholder text:placeholder-type="text">&lt;/for&gt;</text:placeholder></text:p>
@@ -391,4 +391,4 @@
    <text:p text:style-name="P8"><text:placeholder text:placeholder-type="text">&lt;/for&gt;</text:placeholder></text:p>
   </office:text>
  </office:body>
-</office:document>
\ No newline at end of file
+</office:document>
