diff --git a/tryton/trytond/trytond/ir/action.py b/tryton/trytond/trytond/ir/action.py
index 5c83e71449..151a76ee62 100644
--- a/tryton/trytond/trytond/ir/action.py
+++ b/tryton/trytond/trytond/ir/action.py
@@ -1109,7 +1109,7 @@ class ActionWizard(
                     ('model', '=', None),
                     ],
                 ],
-            order=[('model', 'ASC NULLS LAST')])
+            order=[('model', 'ASC NULLS LAST')], limit=1)
         if actions:
             action, = actions
             wiz_name = action.name
