diff --git a/tryton/sao/src/view/tree.js b/tryton/sao/src/view/tree.js
index e1f10c0023f9fffa78aa15f1eb7ea6421e275395_c2FvL3NyYy92aWV3L3RyZWUuanM=..f1a5be22d472234de905d8c42b9191584d6cf7c0_c2FvL3NyYy92aWV3L3RyZWUuanM= 100644
--- a/tryton/sao/src/view/tree.js
+++ b/tryton/sao/src/view/tree.js
@@ -2365,7 +2365,10 @@
                             }
                         }
                         this._get_column_td(next_column)
-                            .find(':input,[tabindex=0]').focus();
+                            .find(':input,[tabindex=0]')
+                            .filter(':visible')
+                            .first()
+                            .focus();
                     } else {
                         var prm = jQuery.when();
                         if (!this.tree.screen.group.parent) {
@@ -2414,4 +2417,5 @@
                                 this._get_column_td(
                                     next_column, next_row)
                                     .trigger('click')
+                                    .trigger('click')
                                     .find(':input,[tabindex=0]')
@@ -2417,4 +2421,6 @@
                                     .find(':input,[tabindex=0]')
+                                    .filter(':visible')
+                                    .first()
                                     .focus();
                             });
                         }
