diff --git a/tryton/trytond/trytond/ir/queue_.py b/tryton/trytond/trytond/ir/queue_.py
index 05b538a62e..9bce2d7396 100644
--- a/tryton/trytond/trytond/ir/queue_.py
+++ b/tryton/trytond/trytond/ir/queue_.py
@@ -247,10 +247,10 @@ class _Method(object):

         if isinstance(instances, list):
             if has_worker and queue_batch:
-                if isinstance(queue_batch, int):
-                    count = queue_batch
-                else:
+                if isinstance(queue_batch, bool):
                     count = batch_size
+                else:
+                    count = int(queue_batch)
             else:
                 count = len(instances)
             task_ids = []
