diff --git a/tryton/trytond/trytond/transaction.py b/tryton/trytond/trytond/transaction.py
index d04350f23f..76f8282ec1 100644
--- a/tryton/trytond/trytond/transaction.py
+++ b/tryton/trytond/trytond/transaction.py
@@ -192,12 +192,12 @@ class Transaction(object):
             self.counter = 0
             self._datamanagers = []
 
+            self.connection = database.get_connection(readonly=readonly,
+                autocommit=autocommit)
             count = 0
             while True:
                 if count:
                     time.sleep(0.002 * (_retry - count))
-                self.connection = database.get_connection(readonly=readonly,
-                    autocommit=autocommit)
                 try:
                     lock_tables = extras.get('_lock_tables', [])
                     for table in lock_tables:
