diff --git a/tryton/sao/src/view/form.js b/tryton/sao/src/view/form.js
index 644dd17558..032dcb717c 100644
--- a/tryton/sao/src/view/form.js
+++ b/tryton/sao/src/view/form.js
@@ -4761,7 +4761,7 @@ function eval_pyson(value){
         },
         set_url: function(value) {
             this.button.attr('href', value);
-            this.button.toggle(value);
+            this.button.toggle(Boolean(value));
         },
         set_invisible: function(invisible) {
             Sao.View.Form.URL._super.set_invisible.call(this, invisible);
