diff --git a/tryton/modules/account/fiscalyear.py b/tryton/modules/account/fiscalyear.py
index 226646b461..6158776b83 100644
--- a/tryton/modules/account/fiscalyear.py
+++ b/tryton/modules/account/fiscalyear.py
@@ -5,7 +5,7 @@ from dateutil.relativedelta import relativedelta
 from sql import Null
 from sql.conditionals import Case
 from sql.operators import Equal, NotEqual
-
+from trytond import backend
 from trytond.cache import Cache
 from trytond.i18n import gettext
 from trytond.model import (
@@ -26,6 +26,7 @@ from .exceptions import (
 STATES = {
     'readonly': Eval('state') != 'open',
 }
+sql_sequence = backend.Database.has_sequence()
 
 
 class FiscalYear(Workflow, ModelSQL, ModelView):
@@ -119,6 +120,7 @@ class FiscalYear(Workflow, ModelSQL, ModelView):
         Sequence = pool.get('ir.sequence')
         SequenceStrict = pool.get('ir.sequence.strict')
 
+        transaction = Transaction()
         table_h = cls.__table_handler__(module)
         cursor = Transaction().connection.cursor()
         t = cls.__table__()
@@ -150,13 +152,22 @@ class FiscalYear(Workflow, ModelSQL, ModelView):
             for sequence_id, in cursor:
                 if sequence_id not in old2new:
                     sequence = Sequence(sequence_id)
+
+                    number_next_internal = sequence.number_increment
+                    if sql_sequence and not sequence._strict:
+                        if transaction.database.sequence_exist(
+                                transaction.connection, sequence._sql_sequence_name):
+                            number_next_internal = transaction.database.sequence_next_number(
+                                transaction.connection, sequence._sql_sequence_name)
+
                     new_sequence = SequenceStrict(
                         name=sequence.name,
+                        company=sequence.company,
                         sequence_type=sequence.sequence_type,
                         prefix=sequence.prefix,
                         suffix=sequence.suffix,
                         type=sequence.type,
-                        number_next_internal=sequence.number_next_internal,
+                        number_next_internal=number_next_internal,
                         number_increment=sequence.number_increment,
                         padding=sequence.padding,
                         timestamp_rounding=sequence.timestamp_rounding,
diff --git a/tryton/modules/account/period.py b/tryton/modules/account/period.py
index 015e6fb041..c80498e697 100644
--- a/tryton/modules/account/period.py
+++ b/tryton/modules/account/period.py
@@ -3,7 +3,7 @@
 
 from sql import Null
 from sql.operators import Equal, NotEqual
-
+from trytond import backend
 from trytond.cache import Cache
 from trytond.const import OPERATORS
 from trytond.i18n import gettext
@@ -21,6 +21,7 @@ from .exceptions import ClosePeriodError, PeriodDatesError, PeriodNotFoundError
 _STATES = {
     'readonly': Eval('state') != 'open',
 }
+sql_sequence = backend.Database.has_sequence()
 
 
 class Period(Workflow, ModelSQL, ModelView):
@@ -112,6 +113,7 @@ class Period(Workflow, ModelSQL, ModelView):
         Sequence = pool.get('ir.sequence')
         SequenceStrict = pool.get('ir.sequence.strict')
 
+        transaction = Transaction()
         table_h = cls.__table_handler__(module)
         cursor = Transaction().connection.cursor()
         t = cls.__table__()
@@ -146,13 +148,21 @@ class Period(Workflow, ModelSQL, ModelView):
             for sequence_id, in cursor:
                 if sequence_id not in old2new:
                     sequence = Sequence(sequence_id)
+
+                    number_next_internal = sequence.number_increment
+                    if sql_sequence and not sequence._strict:
+                        if transaction.database.sequence_exist(
+                                transaction.connection, sequence._sql_sequence_name):
+                            number_next_internal = transaction.database.sequence_next_number(
+                                transaction.connection, sequence._sql_sequence_name)
+
                     new_sequence = SequenceStrict(
                         name=sequence.name,
                         sequence_type=sequence.sequence_type,
                         prefix=sequence.prefix,
                         suffix=sequence.suffix,
                         type=sequence.type,
-                        number_next_internal=sequence.number_next_internal,
+                        number_next_internal=number_next_internal,
                         number_increment=sequence.number_increment,
                         padding=sequence.padding,
                         timestamp_rounding=sequence.timestamp_rounding,
