diff --git a/tryton/trytond/trytond/tests/test_tryton.py b/tryton/trytond/trytond/tests/test_tryton.py
index 78f383c78e..e1e0108111 100644
--- a/tryton/trytond/trytond/tests/test_tryton.py
+++ b/tryton/trytond/trytond/tests/test_tryton.py
@@ -170,19 +170,25 @@ def _sqlite_copy(file_, restore=False):
         finally:
             database.put_connection(connection, True)
 
-    with Transaction().start(DB_NAME, 0) as transaction, \
-            sqlite.connect(file_) as conn2:
-        conn1 = transaction.connection
-        if restore:
-            conn2, conn1 = conn1, conn2
-        if hasattr(conn1, 'backup'):
-            conn1.backup(conn2)
-        else:
-            try:
-                import sqlitebck
-            except ImportError:
-                return False
-            sqlitebck.copy(conn1, conn2)
+    database = backend.Database(DB_NAME).connect()
+    conn1 = database.get_connection()
+    try:
+        with sqlite.connect(file_) as conn2:
+            if restore:
+                conn2, conn1 = conn1, conn2
+            if hasattr(conn1, 'backup'):
+                conn1.backup(conn2)
+            else:
+                try:
+                    import sqlitebck
+                except ImportError:
+                    return False
+                sqlitebck.copy(conn1, conn2)
+        conn2.close()
+
+    finally:
+        database.put_connection(conn1)
+        database.close()
     return True
 
 
