diff --git a/tryton/modules/account_invoice/invoice.py b/tryton/modules/account_invoice/invoice.py
--- a/tryton/modules/account_invoice/invoice.py
+++ b/tryton/modules/account_invoice/invoice.py
@@ -2749,6 +2749,8 @@
             return self.taxes_date
         elif hasattr(self, 'invoice') and hasattr(self.invoice, 'tax_date'):
             return self.invoice.tax_date
+        else:
+            return super().tax_date
 
     def _get_tax_context(self):
         if self.invoice:
diff --git a/tryton/modules/sale_point/sale.py b/tryton/modules/sale_point/sale.py
--- a/tryton/modules/sale_point/sale.py
+++ b/tryton/modules/sale_point/sale.py
@@ -446,7 +446,7 @@
 
     @property
     def tax_date(self):
-        return self.date
+        return self.date or super().tax_date
 
 
 class POSSaleLine(ModelSQL, ModelView, TaxableMixin):
@@ -719,7 +719,7 @@
 
     @property
     def tax_date(self):
-        return self.sale.date
+        return self.sale.date or super().tax_date
 
 
 class POSCashSession(Workflow, ModelSQL, ModelView):
