diff --git a/tryton/trytond/trytond/convert.py b/tryton/trytond/trytond/convert.py
index 91af444e66..79a14f7911 100644
--- a/tryton/trytond/trytond/convert.py
+++ b/tryton/trytond/trytond/convert.py
@@ -768,7 +768,7 @@ class TrytondXmlHandler(sax.handler.ContentHandler):
             fs_values = old_values.copy()
             fs_values.update(new_values)

-            if values != fs_values:
+            if old_values != values or values != fs_values:
                 self.grouped_model_data.extend(([self.ModelData(mdata_id)], {
                             'fs_id': fs_id,
                             'model': model,
