diff --git a/src/view/tree.js b/src/view/tree.js
index d6dfe42838..f01389a81b 100644
--- a/tryton/sao/src/view/tree.js
+++ b/tryton/sao/src/view/tree.js
@@ -790,6 +794,9 @@
 
             // Set column visibility depending on attributes and domain
             var visible_columns = 1;  // start at 1 because of the checkbox
+            if (this.optionals.length && !this.draggable) {
+                visible_columns += 1;
+            }
             var domain = [];
             if (!jQuery.isEmptyObject(this.screen.domain)) {
                 domain.push(this.screen.domain);
@@ -915,7 +922,8 @@
                 }).map(function(row) {
                     return row.el;
                 }));
-                if (this.display_size < this.group.length) {
+                if ((this.display_size < this.group.length) &&
+                    (!this.tbody.children().last().hasClass('more-row'))) {
                     var more_row = jQuery('<tr/>', {
                         'class': 'more-row',
                     });
@@ -928,8 +936,17 @@
                         'title': Sao.i18n.gettext("More"),
                     }).text(Sao.i18n.gettext('More')
                     ).click(() => {
+                        var height = this.table.height();
                         this.display_size += Sao.config.display_size;
                         this.display();
+                        height -= this.treeview.height();
+                        height -= 50;
+                        if (this.tfoot) {
+                            height -= this.tfoot.height();
+                        }
+                        this.treeview[0].scroll({
+                            'top': height,
+                        });
                     });
                     more_cell.append(more_button);
                     more_row.append(more_cell);
