diff --git a/tryton/modules/account_stock_eu/stock.py b/tryton/modules/account_stock_eu/stock.py
index 83c8030b8c..301b1ec287 100644
--- a/tryton/modules/account_stock_eu/stock.py
+++ b/tryton/modules/account_stock_eu/stock.py
@@ -488,7 +488,7 @@ class Move_Production(metaclass=PoolMeta):
         country = super().intrastat_from_country
         if self.from_location:
             if (self.from_location.type == 'production'
-                    and self.production and self.production.warehouse.address):
+                    and self.production and getattr(self.production, 'warehouse', None) and self.production.warehouse.address):
                 country = self.production.warehouse.address.country
         return country
 
@@ -498,7 +498,7 @@ class Move_Production(metaclass=PoolMeta):
         country = super().intrastat_to_country
         if self.to_location:
             if (self.to_location.type == 'production'
-                    and self.production and self.production.warehouse.address):
+                    and self.production and getattr(self.production, 'warehouse', None) and self.production.warehouse.address):
                 country = self.production.warehouse.address.country
         return country
 
