diff --git a/tryton/sao/src/screen.js b/tryton/sao/src/screen.js
index 327a33d59a..934379cde5 100644
--- a/tryton/sao/src/screen.js
+++ b/tryton/sao/src/screen.js
@@ -2286,7 +2286,7 @@
                     return view.display(selected_nodes);
                 } else {
                     var record;
-                    if (!jQuery.isEmptyObject(selected_nodes)) {
+                    if (!jQuery.isEmptyObject(selected_nodes) && !this.current_record) {
                         for (const id of selected_nodes[0]) {
                             const new_record = this.group.get(id);
                             if (!new_record) {
