diff --git a/tryton/modules/account_product/product.py b/tryton/modules/account_product/product.py
index f1b3083e0ff0895c5398a5127109db07d10599f4_bW9kdWxlcy9hY2NvdW50X3Byb2R1Y3QvcHJvZHVjdC5weQ==..53ecc74fc6824f1208a584a877d77bb8703cdf4e_bW9kdWxlcy9hY2NvdW50X3Byb2R1Y3QvcHJvZHVjdC5weQ== 100644
--- a/tryton/modules/account_product/product.py
+++ b/tryton/modules/account_product/product.py
@@ -260,6 +260,12 @@
         else:
             return self.supplier_taxes_deductible_rate
 
+    @classmethod
+    def copy(cls, categories, default=None):
+        default = default.copy() if default is not None else {}
+        default.setdefault('accounting_templates')
+        return super().copy(categories, default=default)
+
 
 class CategoryAccount(ModelSQL, CompanyValueMixin):
     __name__ = 'product.category.account'
