diff --git a/tryton/modules/commission/commission_reporting.py b/tryton/modules/commission/commission_reporting.py
index f147d5a44c..e52fd63dc0 100644
--- a/tryton/modules/commission/commission_reporting.py
+++ b/tryton/modules/commission/commission_reporting.py
@@ -83,7 +83,7 @@ class Agent(ModelView, ModelSQL):
             Round(
                 Sum(Coalesce(commission.base_amount, 0)),
                 currency.digits).as_('base_amount'),
-            Round(Sum(commission.amount), currency.digits).as_('amount'),
+            Round(Sum(commission.amount), price_digits[1]).as_('amount'),
             ]

     @classmethod
