diff -r 1329bfa523a8 trytond/tryton/trytond/modules/analytic_account/line.py
--- a/tryton/modules/analytic_account/line.py   Mon Apr 23 17:28:19 2018 +0200
+++ b/tryton/modules/analytic_account/line.py   Fri May 11 13:16:07 2018 +0200
@@ -29,7 +29,7 @@ class Line(ModelSQL, ModelView):
     account = fields.Many2One(
         'analytic_account.account', "Account", required=True,
         domain=[
-            ('type', 'not in', ['view', 'distribution']),
+            ('type', 'not in', ['view', 'distribution', 'root']),
             ['OR',
                 ('company', '=', None),
                 ('company', '=', Eval('company', -1)),
