--- a/tryton/modules/stock/shipment.py
+++ b/tryton/modules/stock/shipment.py
@@ -2472,6 +2489,8 @@ class ShipmentInternal(ShipmentAssignMixin, Workflow, ModelSQL, ModelView):
                 if m.state != 'done'
                 and m.from_location != shipment.transit_location
                 and m.to_location != shipment.transit_location]
+            if not moves:
+                continue
             Move.copy(moves, default={
                     'to_location': shipment.transit_location.id,
                     'planned_date': shipment.planned_start_date,
