diff --git a/tryton/trytond/trytond/modules/analytic_account/account.py b/tryton/trytond/trytond/modules/analytic_account/account.py
index ef063970ec..b2fa817eb7 100644
--- a/tryton/trytond/trytond/modules/analytic_account/account.py
+++ b/tryton/trytond/trytond/modules/analytic_account/account.py
@@ -143,7 +143,7 @@ class Account(
     @classmethod
     def check_move_domain(cls, accounts, field_names):
         pool = Pool()
-        Line = pool.get('account.move.line')
+        Line = pool.get('analytic_account.line')
         if field_names and 'type' not in field_names:
             return
         accounts = [
