diff --git a/tryton/sao/src/common.js b/tryton/sao/src/common.js
index 8d8b5e4084..92fccb3ba8 100644
--- a/tryton/sao/src/common.js
+++ b/tryton/sao/src/common.js
@@ -2014,8 +2014,11 @@
                 }
                 let { format } = new Intl.NumberFormat(
                     Sao.i18n.BC47(Sao.i18n.getlang()));
+                let [, thousandSeparator] = /^1(.)?000/.exec(format(1000));
                 let [, decimalSign] = /^0(.)1$/.exec(format(0.1));
-                return Number(string);
+                return Number(string
+                    .replaceAll(thousandSeparator, '')
+                    .replace(decimalSign, '.'));
             }
             var convert_selection = function() {
                 if (typeof value == 'string') {
