diff --git a/tryton/modules/purchase/purchase.py b/tryton/modules/purchase/purchase.py
index 0d64f765df..587b558ff0 100644
--- a/tryton/modules/purchase/purchase.py
+++ b/tryton/modules/purchase/purchase.py
@@ -1,6 +1,7 @@
 # This file is part of Tryton.  The COPYRIGHT file at the top level of
 # this repository contains the full copyright notices and license terms.
 import datetime
+import math
 from collections import defaultdict
 from decimal import Decimal
 from itertools import chain, groupby
@@ -1338,7 +1339,8 @@ class Line(sequence_ordered(), ModelSQL, ModelView):
             for move in self.moves:
                 if move in moves_ignored:
                     quantity -= UoM.compute_qty(
-                        move.unit, move.quantity, self.unit)
+                        move.unit, math.copysign(move.quantity, self.quantity),
+                        self.unit)
         for invoice_line in self.invoice_lines:
             if invoice_line.type != 'line':
                 continue
diff --git a/tryton/modules/sale/sale.py b/tryton/modules/sale/sale.py
index 9c06fc5a50..f416669d91 100644
--- a/tryton/modules/sale/sale.py
+++ b/tryton/modules/sale/sale.py
@@ -1,6 +1,7 @@
 # This file is part of Tryton.  The COPYRIGHT file at the top level of
 # this repository contains the full copyright notices and license terms.
 import datetime
+import math
 from collections import defaultdict
 from decimal import Decimal
 from functools import partial
@@ -1415,7 +1416,8 @@ class SaleLine(TaxableMixin, sequence_ordered(), ModelSQL, ModelView):
             for move in self.moves:
                 if move in moves_ignored:
                     quantity -= UoM.compute_qty(
-                        move.unit, move.quantity, self.unit)
+                        move.unit, math.copysign(move.quantity, self.quantity),
+                        self.unit)
         for invoice_line in self.invoice_lines:
             if invoice_line.type != 'line':
                 continue
