diff --git a/tryton/sao/src/tab.js b/tryton/sao/src/tab.js
index 4281600d1f..b67ce4ec06 100644
--- a/tryton/sao/src/tab.js
+++ b/tryton/sao/src/tab.js
@@ -1249,7 +1249,7 @@
                 this.refresh_resources(true);
             });
             for (const file of files) {
-                Sao.common.get_file_data(file, window_.add_data);
+                Sao.common.get_file_data(file, window_.add_data.bind(window_));
             }
             jQuery.when.apply(jQuery, uris).then(function() {
                 function empty(value) {
