diff --git a/tryton/sao/src/model.js b/tryton/sao/src/model.js
index 8fa2b3f4ea..7569268750 100644
--- a/tryton/sao/src/model.js
+++ b/tryton/sao/src/model.js
@@ -718,7 +718,8 @@
                     ((fdescription.loading || 'lazy') == 'eager')) {
                     fnames_to_fetch.push(fname + ':string');
                 } else if (
-                    ['many2many', 'one2many'].includes(fdescription.type)) {
+                    ['many2many', 'one2many'].includes(fdescription.type) &&
+                    (loading == 'lazy')) {
                     var sub_fields = get_x2m_sub_fields(fdescription, fname);
                     fnames_to_fetch = [ ...fnames_to_fetch, ...sub_fields];
                     if (sub_fields.length > 0) {
diff --git a/tryton/trytond/trytond/model/modelsql.py b/tryton/trytond/trytond/model/modelsql.py
index f28c6dfbee..2b368ff2a7 100644
--- a/tryton/trytond/trytond/model/modelsql.py
+++ b/tryton/trytond/trytond/model/modelsql.py
@@ -1229,6 +1229,8 @@ class ModelSQL(ModelStorage):
                 if value is not None:
                     add(value)
             related_read_limit = transaction.context.get('related_read_limit')
+            # use dict keys to make target ids unique but preserving the order
+            target_ids = list(dict.fromkeys(target_ids).keys())
             rows = Target.read(target_ids[:related_read_limit], fields)
             if related_read_limit is not None:
                 rows += [{'id': i} for i in target_ids[related_read_limit:]]
