diff --git a/tryton/modules/stock_lot/stock.py b/tryton/modules/stock_lot/stock.py
index d5fac5e16d..ed9d2df506 100644
--- a/tryton/modules/stock_lot/stock.py
+++ b/tryton/modules/stock_lot/stock.py
@@ -485,7 +485,7 @@ class Move(metaclass=PoolMeta):
                 moves, with_childs=with_childs, grouping=grouping, pblc=pblc)
         return success
 
-    @fields.depends('product', 'lot')
+    @fields.depends('product', 'lot', '_parent_lot.product')
     def on_change_product(self):
         try:
             super().on_change_product()
