diff --git a/tryton/sao/src/sao.less b/tryton/sao/src/sao.less
index d0abbc0b25..e8f0b1ec35 100644
--- a/tryton/sao/src/sao.less
+++ b/tryton/sao/src/sao.less
@@ -511,6 +511,18 @@ img.icon {
     display: flex;
     flex-direction: column;
     flex: 1;
+    position: relative;
+}
+
+.overlay-mask {
+    background: black;
+    bottom: 0;
+    left: 0;
+    opacity: 5%;
+    position: absolute;
+    right: 0;
+    top: 0;
+    z-index: 500;
 }

 .filter-box {
diff --git a/tryton/sao/src/screen.js b/tryton/sao/src/screen.js
index e6e230a8ef..a1de530683 100644
--- a/tryton/sao/src/screen.js
+++ b/tryton/sao/src/screen.js
@@ -17,6 +17,9 @@
             this.el = jQuery('<div/>', {
                 'class': 'screen-container'
             });
+            this.overlay_mask = jQuery('<div/>', {
+                'class': 'overlay-mask',
+            }).appendTo(this.el);
             this.filter_box = jQuery('<form/>', {
                 'class': 'filter-box hidden-xs'
             }).submit(e => {
@@ -220,6 +223,12 @@
             }
             this.el.append(this.content_box);
         },
+        disable: function() {
+            this.overlay_mask.show();
+        },
+        enable: function() {
+            this.overlay_mask.hide();
+        },
         set_text: function(value) {
             this.search_entry.val(value);
             this.bookmark_match();
@@ -1360,6 +1369,7 @@
                 () => this.set_tree_state().then(() => {
                     var record = this.current_record
                     this.current_record = record;
+                    this.screen_container.enable();
                     // set_cursor must be called after set_tree_state because
                     // set_tree_state redraws the tree
                     if (set_cursor) {
@@ -1585,13 +1595,14 @@
             }
         },
         cancel_current: function(initial_value) {
+            this.screen_container.disable();
             var prms = [];
             if (this.current_record) {
                 this.current_record.cancel();
                 if (this.current_record.id < 0) {
                     if (initial_value) {
                         this.current_record.reset(initial_value);
-                        this.display();
+                        prms.push(this.display());
                     } else {
                         prms.push(this.remove(
                             false, false, false, [this.current_record]));
@@ -1616,6 +1627,7 @@
                 this.current_view.set_value();
                 var fields = this.current_view.get_fields();
             }
+            this.screen_container.disable();
             var path = current_record.get_path(this.group);
             var prm = jQuery.Deferred();
             if (this.current_view && (this.current_view.view_type == 'tree')) {
@@ -1922,6 +1934,7 @@
             return this.current_record.get_on_change_value();
         },
         reload: function(ids, written) {
+            this.screen_container.disable();
             this.group.reload(ids);
             var promises = [];
             if (written) {
