diff --git a/tryton/sao/src/model.js b/tryton/sao/src/model.js
index 3954d9471b..48e463b843 100644
--- a/tryton/sao/src/model.js
+++ b/tryton/sao/src/model.js
@@ -1352,6 +1352,9 @@
             }
         },
         get_loaded: function(fields) {
+            if (this.id < 0) {
+                return true;
+            }
             if (!fields) {
                 fields = Object.keys(this.model.fields);
             }
