diff --git a/tryton/sao/src/common.js b/tryton/sao/src/common.js
index eff2febd8f..046600c970 100644
--- a/tryton/sao/src/common.js
+++ b/tryton/sao/src/common.js
@@ -2019,13 +2019,8 @@
                 }
                 let { format } = new Intl.NumberFormat(
                     Sao.i18n.BC47(Sao.i18n.getlang()));
-                // use 10000 because some language (ex: es) add thousand
-                // separator only after 9999
-                let [, thousandSeparator] = /^10(.)?000/.exec(format(10000));
                 let [, decimalSign] = /^0(.)1$/.exec(format(0.1));
-                return Number(string
-                    .replace(new RegExp(thousandSeparator, 'g'), '')
-                    .replace(decimalSign, '.'));
+                return Number(string.replace(decimalSign, '.'));
             }
             var convert_selection = function() {
                 if (typeof value == 'string') {
