diff --git a/tryton/sao/src/wizard.js b/tryton/sao/src/wizard.js
index 72a029045f..ea14bdfd5d 100644
--- a/tryton/sao/src/wizard.js
+++ b/tryton/sao/src/wizard.js
@@ -316,7 +316,7 @@
                 }
                 if (!dialog ||
                     !this.model ||
-                    (Sao.main_menu_screen.model_name == this.model)) {
+                    (Sao.main_menu_screen && (Sao.main_menu_screen.model_name == this.model))) {
                     is_menu = true;
                     screen = Sao.main_menu_screen;
                 } else {
