diff --git a/tryton/modules/purchase_request/purchase.py b/tryton/modules/purchase_request/purchase.py
index 63425351b4..a63b557763 100644
--- a/tryton/modules/purchase_request/purchase.py
+++ b/tryton/modules/purchase_request/purchase.py
@@ -91,6 +91,12 @@ class Line(metaclass=PoolMeta):
             'invisible': ~Eval('requests'),
             })
 
+    @classmethod
+    def copy(cls, lines, default=None):
+        default = default.copy() if default is not None else {}
+        default.setdefault('requests')
+        return super().copy(lines, default=default)
+
     @classmethod
     def delete(cls, lines):
         pool = Pool()
