diff --git a/tryton/modules/production_split/production.py b/tryton/modules/production_split/production.py
index 4340cbefe5..7100b141e6 100644
--- a/tryton/modules/production_split/production.py
+++ b/tryton/modules/production_split/production.py
@@ -55,6 +55,7 @@ class Production(metaclass=PoolMeta):
                 count -= 1
         if n_productions:
             productions.extend(self.copy([self] * n_productions, {
+                        'reference': lambda data: data['reference'],
                         'quantity': quantity,
                         'unit': unit.id,
                         'inputs': None,
diff --git a/tryton/modules/stock_split/stock.py b/tryton/modules/stock_split/stock.py
index 69eb004660..6c90f9003c 100644
--- a/tryton/modules/stock_split/stock.py
+++ b/tryton/modules/stock_split/stock.py
@@ -230,7 +230,12 @@ class SplitShipment(Wizard):
             Move.delete(
                 [m for m in shipment.inventory_moves if m.state == 'draft'])
 
-        shipment, = Shipment.copy([shipment], default={'moves': None})
+        shipment, = Shipment.copy(
+            [shipment],
+            default={
+                'reference': lambda data: data['reference'],
+                'moves': None,
+                })
         Move.write(list(self.start.moves), {'shipment': str(shipment)})
         return 'end'
 
