diff -r 6597f33e51d2 invoice.py
--- a/trytond/trytond/modules/account_invoice/invoice.py	Fri Aug 09 16:33:39 2019 +0200
+++ b/trytond/trytond/modules/account_invoice/invoice.py	Wed Aug 28 23:44:50 2019 +0200
@@ -2657,7 +2657,7 @@ class InvoiceTax(sequence_ordered(), ModelSQL, ModelView):
                 self.account = self.tax.credit_note_account
 
     @fields.depends(
-        'tax', 'base', 'amount', 'manual', 'invoice',
+        'tax', 'base', 'manual', 'invoice',
         '_parent_invoice.currency',
         # From_date
         '_parent_invoice.accounting_date', '_parent_invoice.invoice_date',
@@ -2677,7 +2677,6 @@ class InvoiceTax(sequence_ordered(), ModelSQL, ModelView):
                         if self.invoice.currency:
                             amount = self.invoice.currency.round(amount)
                         return amount
-        return self.amount
 
     @property
     def _key(self):
