diff --git a/tryton/trytond/trytond/application.py b/tryton/trytond/trytond/application.py
index 3c7103539f52640379094ac99ffc5c65f4953943_dHJ5dG9uZC90cnl0b25kL2FwcGxpY2F0aW9uLnB5..63609a232127f6a219c5bcd5c2dc8086d5db01e9_dHJ5dG9uZC90cnl0b25kL2FwcGxpY2F0aW9uLnB5 100644
--- a/tryton/trytond/trytond/application.py
+++ b/tryton/trytond/trytond/application.py
@@ -31,7 +31,12 @@
 # TRYTOND_CONFIG it's managed by importing config
 db_names = os.environ.get('TRYTOND_DATABASE_NAMES')
 if db_names:
+    from trytond import backend
+
+    def initializer(name):
+        Pool(name).init()
+        backend.Database(name).close()
     # Read with csv so database name can include special chars
     reader = csv.reader(StringIO(db_names))
     threads = []
     for name in next(reader):
@@ -34,8 +39,8 @@
     # Read with csv so database name can include special chars
     reader = csv.reader(StringIO(db_names))
     threads = []
     for name in next(reader):
-        thread = threading.Thread(target=lambda: Pool(name).init())
+        thread = threading.Thread(target=initializer, args=(name,))
         thread.start()
         threads.append(thread)
     for thread in threads:
diff --git a/trytond/trytond/backend/postgresql/database.py b/trytond/trytond/backend/postgresql/database.py
index 3c7103539f52640379094ac99ffc5c65f4953943_dHJ5dG9uZC90cnl0b25kL2JhY2tlbmQvcG9zdGdyZXNxbC9kYXRhYmFzZS5weQ==..63609a232127f6a219c5bcd5c2dc8086d5db01e9_dHJ5dG9uZC90cnl0b25kL2JhY2tlbmQvcG9zdGdyZXNxbC9kYXRhYmFzZS5weQ== 100644
--- a/trytond/trytond/backend/postgresql/database.py
+++ b/trytond/trytond/backend/postgresql/database.py
@@ -286,7 +286,7 @@
     def close(self):
         with self._lock:
             logger.info('disconnection from "%s"', self.name)
-            self._connpool.close()
+            self._connpool.close(None)
             self._databases[os.getpid()].pop(self.name)
 
     @classmethod
