diff --git a/tryton/modules/stock/move.py b/tryton/modules/stock/move.py
index 7cf384510d..807f1c3313 100644
--- a/tryton/modules/stock/move.py
+++ b/tryton/modules/stock/move.py
@@ -801,8 +801,10 @@ class Move(Workflow, ModelSQL, ModelView):
         # Use ordered dict to optimize cache alignment
         products, uoms = OrderedDict(), OrderedDict()
         for vals in vlist:
-            products[vals['product']] = None
-            uoms[vals['uom']] = None
+            if vals.get('product') is not None:
+                products[vals['product']] = None
+            if vals.get('uom') is not None:
+                uoms[vals['uom']] = None
         id2product = {p.id: p for p in Product.browse(products.keys())}
         id2uom = {u.id: u for u in Uom.browse(uoms.keys())}
         for vals in vlist:
