diff --git a/trytond/trytond/modules/account/move.py b/trytond/trytond/modules/account/move.py
index 28ea9b8..63b4f6f 100644
--- a/trytond/trytond/modules/account/move.py
+++ b/trytond/trytond/modules/account/move.py
@@ -1759,17 +1759,18 @@ class Reconcile(Wizard):
             self.show.parties = parties
             return party,

-        if getattr(self.show, 'accounts', None) is None:
-            self.show.accounts = self.get_accounts()
-            if not next_account():
-                return 'end'
-        if getattr(self.show, 'parties', None) is None:
-            self.show.parties = self.get_parties(self.show.account)
-
-        while not next_party():
-            if not next_account():
-                return 'end'
-        return 'show'
+        with Transaction().set_context(_check_access=True):
+            if getattr(self.show, 'accounts', None) is None:
+                self.show.accounts = self.get_accounts()
+                if not next_account():
+                    return 'end'
+            if getattr(self.show, 'parties', None) is None:
+                self.show.parties = self.get_parties(self.show.account)
+
+            while not next_party():
+                if not next_account():
+                    return 'end'
+            return 'show'

     def default_show(self, fields):
         pool = Pool()
