diff --git a/tryton/modules/account_payment_clearing/payment.py b/tryton/modules/account_payment_clearing/payment.py
index ff158f0c75..5eb0a1b23a 100644
--- a/tryton/modules/account_payment_clearing/payment.py
+++ b/tryton/modules/account_payment_clearing/payment.py
@@ -397,7 +397,7 @@ class Group(metaclass=PoolMeta):
         Payment = pool.get('account.payment')
         payment = Payment.__table__()
         cursor = Transaction().connection.cursor()
-        result = defaultdict()
+        result = defaultdict(lambda: None)
         column = Coalesce(payment.clearing_reconciled, False)
         if backend.name == 'sqlite':
             column = Min(column)
