diff --git a/tryton/modules/purchase_request_quotation/purchase.py b/tryton/modules/purchase_request_quotation/purchase.py
index 2923966586..05c9b8101a 100644
--- a/tryton/modules/purchase_request_quotation/purchase.py
+++ b/tryton/modules/purchase_request_quotation/purchase.py
@@ -226,7 +226,6 @@ class Quotation(Workflow, ModelSQL, ModelView):
             default = default.copy()
         default.setdefault('number', None)
         default.setdefault('revision', cls.default_revision())
-        default.setdefault('request_quotations', None)
         return super(Quotation, cls).copy(groups, default=default)

     @property
