diff --git a/tryton/modules/sale_shipment_cost/sale.py b/tryton/modules/sale_shipment_cost/sale.py
index d05ce678fd..0e1d11d920 100644
--- a/tryton/modules/sale_shipment_cost/sale.py
+++ b/tryton/modules/sale_shipment_cost/sale.py
@@ -317,8 +317,12 @@ class Sale(metaclass=PoolMeta):
         return cost_line

     def _get_shipment_grouping_fields(self, shipment):
-        return super()._get_shipment_grouping_fields(shipment) + [
-            'cost_method', 'carrier']
+        pool = Pool()
+        ShipmentOut = pool.get('stock.shipment.out')
+        fields = super()._get_shipment_grouping_fields(shipment) + ['carrier']
+        if isinstance(shipment, ShipmentOut):
+            fields.append('cost_method')
+        return fields

     @property
     def shipment_cost_amount(self):
