diff --git a/invoice.py b/invoice.py
index 582265d..1093bb3 100644
--- a/trytond/trytond/modules/account_invoice/invoice.py
+++ b/trytond/trytond/modules/account_invoice/invoice.py
@@ -1202,7 +1202,7 @@ class Invoice(Workflow, ModelSQL, ModelView, TaxableMixin):
                     gettext('account_invoice.msg_invoice_same_account_line',
                         account=self.account.rec_name,
                         invoice=self.rec_name,
-                        line=line.rec_name))
+                        lines=line.rec_name))
 
     def check_payment_lines(self):
         amount = sum(l.debit - l.credit for l in self.lines_to_pay)
@@ -2121,7 +2121,7 @@ class InvoiceLine(sequence_ordered(), ModelSQL, ModelView, TaxableMixin):
                     gettext('account_invoice.msg_invoice_same_account_line',
                         account=self.account.rec_name,
                         invoice=self.invoice.rec_name,
-                        line=self.rec_name))
+                        lines=self.rec_name))
 
     def _compute_taxes(self):
         pool = Pool()
