diff --git a/tryton/modules/purchase/purchase.py b/tryton/modules/purchase/purchase.py
index 0bc42b126a..9d4a6a4a20 100644
--- a/tryton/modules/purchase/purchase.py
+++ b/tryton/modules/purchase/purchase.py
@@ -1333,7 +1333,7 @@ class Line(sequence_ordered(), ModelSQL, ModelView):
                 and self.product_supplier not in product_suppliers):
             self.product_supplier = None

-            self.unit_price = self.compute_unit_price()
+        self.unit_price = self.compute_unit_price()

         self.amount = self.on_change_with_amount()

diff --git a/tryton/modules/sale/sale.py b/tryton/modules/sale/sale.py
index f412dbc85b..32c65db2fd 100644
--- a/tryton/modules/sale/sale.py
+++ b/tryton/modules/sale/sale.py
@@ -1328,7 +1328,7 @@ class SaleLine(TaxableMixin, sequence_ordered(), ModelSQL, ModelView):
             self.unit = self.product.sale_uom
             self.unit_digits = self.product.sale_uom.digits

-            self.unit_price = self.compute_unit_price()
+        self.unit_price = self.compute_unit_price()

         self.amount = self.on_change_with_amount()
