diff --git a/tryton/sao/src/view/form.js b/tryton/sao/src/view/form.js
index c40c1c4348..7812dd92d9 100644
--- a/tryton/sao/src/view/form.js
+++ b/tryton/sao/src/view/form.js
@@ -2203,14 +2203,14 @@ function eval_pyson(value){
             }
         },
         get width() {
-            var digits = this.digits;
-            if (digits && digits.every(d => d !== null)) {
-                return digits.reduce(function(acc, cur) {
-                    return acc + cur;
-                });
-            } else {
-                return this.attributes.width || 18;
-            }
+            let default_ = [16, 2];
+            let digits = this.digits || [null, null];
+            return digits.reduce((acc, cur, i) => {
+                if (cur === null) {
+                    cur = default_[i];
+                }
+                return acc + cur;
+            }, 0);
         },
         display: function() {
             var record = this.record;
