diff --git a/tryton/modules/sale_opportunity/opportunity.py b/tryton/modules/sale_opportunity/opportunity.py
index b195e4f2ba..5ebf9d6328 100644
--- a/tryton/modules/sale_opportunity/opportunity.py
+++ b/tryton/modules/sale_opportunity/opportunity.py
@@ -575,12 +575,18 @@ class SaleOpportunityLine(sequence_ordered(), ModelSQL, ModelView):
                 ('salable', '=', True),
                 ()),
             ],
+        context={
+            'company': Eval('company', None),
+            },
+        depends={'company'},
         states=_states)
     quantity = fields.Float(
         "Quantity", digits='unit', required=True, states=_states)
     unit = fields.Many2One('product.uom', 'Unit', required=True,
         states=_states)
-
+    company = fields.Function(
+        fields.Many2One('company.company', "Company"),
+        'on_change_with_company')
     del _states

     @classmethod
@@ -599,6 +605,10 @@ class SaleOpportunityLine(sequence_ordered(), ModelSQL, ModelView):
         if self.opportunity:
             return self.opportunity.state

+    @fields.depends('opportunity', '_parent_opportunity.company')
+    def on_change_with_company(self, name=None):
+        return self.opportunity.company if self.opportunity else None
+
     @fields.depends('product', 'unit')
     def on_change_product(self):
         if not self.product:
