diff --git a/tryton/sao/src/model.js b/tryton/sao/src/model.js
index 81e8639d9e..8b55441294 100644
--- a/tryton/sao/src/model.js
+++ b/tryton/sao/src/model.js
@@ -2071,6 +2071,10 @@
                     rec_name = '';
                 }
             }
+            if (value < 0) && (this.name != record.group.parent_name)) {
+                value = null;
+                rec_name = '';
+            }
             Sao.setdefault(
                 record._values, this.name + '.', {}).rec_name = rec_name;
             Sao.field.Many2One._super.set_client.call(this, record, value,
