diff --git a/tryton/modules/account_invoice/invoice.py b/tryton/modules/account_invoice/invoice.py
index bf7d348d24..f40b3be375 100644
--- a/tryton/modules/account_invoice/invoice.py
+++ b/tryton/modules/account_invoice/invoice.py
@@ -1577,7 +1577,7 @@ class Invoice(Workflow, ModelSQL, ModelView, TaxableMixin, InvoiceReportMixin):
 
     def compute_fields(self, field_names=None):
         values = super().compute_fields(field_names=field_names)
-        if (not field_names or 'number' in field_names):
+        if (field_names is None or 'number' in field_names):
             values['number_alnum'] = (
                 re.sub(r'[\W_]', '', self.number)
                 if self.number is not None else None)
