diff --git a/tryton/trytond/trytond/ir/ui/view.py b/tryton/trytond/trytond/ir/ui/view.py
index a5c3a327d7..26e30a1070 100644
--- a/tryton/trytond/trytond/ir/ui/view.py
+++ b/tryton/trytond/trytond/ir/ui/view.py
@@ -354,7 +354,7 @@ class View(
                 position = element.get('position', 'inside')
                 new_tree = getattr(cls, '_inherit_apply_%s' % position)(
                     tree, element, target)
-                if new_tree:
+                if new_tree is not None:
                     tree = new_tree
         return tree
 
