diff --git a/trytond/trytond/modules/stock/period.py b/trytond/trytond/modules/stock/period.py
index 54e5f4b..a411320 100644
--- a/trytond/trytond/modules/stock/period.py
+++ b/trytond/trytond/modules/stock/period.py
@@ -152,7 +152,9 @@ class Period(Workflow, ModelSQL, ModelView):
                         values[field] = key[i]
                     to_create.append(values)
             if to_create:
-                Cache.create(to_create)
+                to_create = [x for x in to_create if x['internal_quantity']]
+                for sub_to_create in grouped_slice(to_create):
+                    Cache.create(list(sub_to_create))
 
 
 class Cache(ModelSQL, ModelView):
