diff --git a/tryton/sao/src/view/tree.js b/tryton/sao/src/view/tree.js
index 3ae7a32ba3..995b876718 100644
--- a/tryton/sao/src/view/tree.js
+++ b/tryton/sao/src/view/tree.js
@@ -2561,12 +2561,18 @@
             cell = cell.children('a');
             cell.unbind('click');
             Sao.View.Tree.Many2OneColumn._super.update_text.call(this, cell, record);
+            let view_ids = (this.attributes.view_ids || '').split(',');
+            if (!jQuery.isEmptyObject(view_ids)) {
+                // Remove the first tree view as mode is form only
+                view_ids.shift();
+            }
             cell.click(event => {
                 event.stopPropagation();
                 var params = {};
                 params.model = this.attributes.relation;
                 params.res_id = this.field.get(record);
                 params.mode = ['form'];
+                params.view_ids = view_ids;
                 params.name = this.attributes.string;
                 params.context = this.field.get_context(record);
                 Sao.Tab.create(params);
