diff --git a/modules/account_payment_sepa/payment.py b/modules/account_payment_sepa/payment.py
--- a/tryton/modules/account_payment_sepa/payment.py
+++ b/tryton/modules/account_payment_sepa/payment.py
@@ -867,7 +867,10 @@
         }
     message = fields.Binary('Message', filename='filename',
         file_id=file_id, store_prefix=store_prefix,
-        states=_states)
+        states={
+            'required': True,
+            'readonly': _states['readonly'],
+            })
     message_file_id = fields.Char("Message File ID", readonly=True)
     filename = fields.Function(fields.Char('Filename'), 'get_filename')
     type = fields.Selection([
