diff --git a/tryton/trytond/trytond/backend/postgresql/database.py b/tryton/trytond/trytond/backend/postgresql/database.py
index 74f84aeb1225f12d88cbbb3e04455065932973b6_dHJ5dG9uZC90cnl0b25kL2JhY2tlbmQvcG9zdGdyZXNxbC9kYXRhYmFzZS5weQ==..25df99fab0c687a00a2566a7c9932f5b1c06989b_dHJ5dG9uZC90cnl0b25kL2JhY2tlbmQvcG9zdGdyZXNxbC9kYXRhYmFzZS5weQ== 100644
--- a/tryton/trytond/trytond/backend/postgresql/database.py
+++ b/tryton/trytond/trytond/backend/postgresql/database.py
@@ -19,7 +19,7 @@
 from psycopg import connect, rows
 from psycopg.errors import QueryCanceled as DatabaseTimeoutError
 from psycopg.sql import SQL, Identifier
-from psycopg_pool import ConnectionPool
+from psycopg_pool import ConnectionPool, NullConnectionPool
 from sql import Cast, Flavor, For, Literal, Table
 from sql.aggregate import Count
 from sql.conditionals import Coalesce
@@ -221,6 +221,8 @@
 
         minconn = config.getint('database', 'minconn', default=1)
         maxconn = config.getint('database', 'maxconn', default=64)
+        if name == _default_name:
+            minconn = 0
         timeout = config.getint('database', 'timeout')
         last_clean = (now - cls._clean_last).total_seconds()
         if last_clean > timeout:
@@ -236,4 +238,8 @@
                     kwargs = cls._connection_params(name)
                     kwargs['cursor_factory'] = LoggingCursor
                     conninfo = kwargs.pop('conninfo')
+                    if name == _default_name:
+                        Pool = NullConnectionPool
+                    else:
+                        Pool = ConnectionPool
                     try:
@@ -239,5 +245,5 @@
                     try:
-                        inst._connpool = ConnectionPool(
+                        inst._connpool = Pool(
                             conninfo,
                             kwargs=kwargs,
                             open=True,
@@ -241,7 +247,7 @@
                             conninfo,
                             kwargs=kwargs,
                             open=True,
-                            check=ConnectionPool.check_connection,
+                            check=Pool.check_connection,
                             min_size=minconn, max_size=maxconn)
                     except Exception:
                         logger.error(
