diff --git a/tryton/modules/sale_advance_payment/sale.py b/tryton/modules/sale_advance_payment/sale.py
index c1de57f355..5b6778f0cc 100644
--- a/tryton/modules/sale_advance_payment/sale.py
+++ b/tryton/modules/sale_advance_payment/sale.py
@@ -384,6 +384,11 @@ class Sale(metaclass=PoolMeta):
         recalled_lines = {
             k: list(v) for k, v in groupby(recalled_lines, lambda l: l.origin)}
         for advance_line in advance_lines:
+            already_used = InvoiceLine.search([
+                ('origin', '=', str(advance_line)),
+                ], limit=1)
+            if already_used:
+                continue
             amount = advance_line.amount
             if advance_line in recalled_lines:
                 for recalled_line in recalled_lines[advance_line]:
