diff --git a/tryton/modules/account/move.py b/tryton/modules/account/move.py
index 123755b2e9..564a9dc7db 100644
--- a/tryton/modules/account/move.py
+++ b/tryton/modules/account/move.py
@@ -946,7 +946,7 @@ class Line(MoveLineMixin, ModelSQL, ModelView):
     def on_change_move(self):
         if self.move:
             if not self.debit and not self.credit:
-                total = sum(l.debit - l.credit
+                total = sum((l.debit or 0) - (l.credit or 0)
                     for l in getattr(self.move, 'lines', []))
                 self.debit = -total if total < 0 else Decimal(0)
                 self.credit = total if total > 0 else Decimal(0)
@@ -1014,7 +1014,8 @@ class Line(MoveLineMixin, ModelSQL, ModelView):
         'Set correct sign to amount_second_currency'
         if self.amount_second_currency:
             self.amount_second_currency = \
-                self.amount_second_currency.copy_sign(self.debit - self.credit)
+                self.amount_second_currency.copy_sign(
+                    (self.debit or 0) - (self.credit or 0))

     @fields.depends('account')
     def on_change_account(self):
