diff --git a/tryton/sao/src/view/form.js b/tryton/sao/src/view/form.js
index 7c291d6c5a..50167d704a 100644
--- a/tryton/sao/src/view/form.js
+++ b/tryton/sao/src/view/form.js
@@ -872,7 +872,7 @@ function eval_pyson(value){
         hide: function() {
             Sao.View.Form.Page._super.hide.call(this);
             if (this.el.hasClass('active')) {
-                this.el.next(':visible').find('a').tab('show');
+                this.el.siblings(':visible').first().find('a').tab('show');
             }
         }
     });
