diff --git a/tryton/modules/sale_promotion/sale.py b/tryton/modules/sale_promotion/sale.py
index 167cdd011c..5e487e8279 100644
--- a/tryton/modules/sale_promotion/sale.py
+++ b/tryton/modules/sale_promotion/sale.py
@@ -115,8 +115,8 @@ class Line(metaclass=PoolMeta):
 
     original_unit_price = Monetary(
         "Original Unit Price", digits=price_digits, currency='currency',
-        readonly=True,
         states={
+            'editable': False,
             'required': Bool(Eval('promotion', None)),
             'invisible': ~Eval('promotion'),
             })
