diff --git a/tryton/trytond/trytond/model/modelstorage.py b/tryton/trytond/trytond/model/modelstorage.py
index 759768ce06..c78cce351b 100644
--- a/tryton/trytond/trytond/model/modelstorage.py
+++ b/tryton/trytond/trytond/model/modelstorage.py
@@ -2048,8 +2048,8 @@ class ModelStorage(Model):
                         cls.write(*sum(
                                 (([r], save_values[r]) for r in to_write), ()))
             except Exception:
-                for record in to_create + to_write:
-                    record._values = values[record]
+                for record in chain(to_create, to_write):
+                    record._values = values.get(record)
                 raise
             for record in to_create + to_write:
                 record._init_values = None
