diff --git a/tryton/sao/src/view/form.js b/tryton/sao/src/view/form.js
index 41e35a8028..29afa16781 100644
--- a/tryton/sao/src/view/form.js
+++ b/tryton/sao/src/view/form.js
@@ -5568,8 +5568,7 @@ function eval_pyson(value){
             return this._parse(this.format, this.input.val());
         },
         set_value: function(value) {
-            if ((value instanceof Sao.DateTime) ||
-                (value instanceof Sao.Date)) {
+            if (value && (value.isDate || value.isDateTime)) {
                 value = this._format(this.format, value);
             } else {
                 value = '';
