diff --git a/tryton/modules/account/account.py b/tryton/modules/account/account.py
index db6fe659e5..ec0f7a4cf0 100644
--- a/tryton/modules/account/account.py
+++ b/tryton/modules/account/account.py
@@ -1348,7 +1348,7 @@ class Account(
         childs = [self]
         while childs:
             for child in childs:
-                if not child.template:
+                if not child.template or child.template_override:
                     continue
                 values = {}
                 tax_ids = [template2tax[x.id] for x in child.template.taxes
