diff --git a/trytond/trytond/modules/product/uom.py b/trytond/trytond/modules/product/uom.py
index e07c8c4..dd67c55 100644
--- a/trytond/trytond/modules/product/uom.py
+++ b/trytond/trytond/modules/product/uom.py
@@ -7,7 +7,7 @@ from math import ceil, floor, log10
 from trytond.config import config
 from trytond.i18n import gettext
 from trytond.model import (
-    ModelView, ModelSQL, DeactivableMixin, fields, Check, SymbolMixin)
+    ModelView, ModelSQL, DeactivableMixin, fields, Check, SymbolMixin, DigitsMixin)
 from trytond.model.exceptions import AccessError
 from trytond.pyson import Eval
 from trytond.transaction import Transaction
@@ -32,7 +32,7 @@ class UomCategory(ModelSQL, ModelView):
         cls._order.insert(0, ('name', 'ASC'))


-class Uom(SymbolMixin, DeactivableMixin, ModelSQL, ModelView):
+class Uom(SymbolMixin, DigitsMixin, DeactivableMixin, ModelSQL, ModelView):
     "Unit of Measure"
     __name__ = 'product.uom'
     name = fields.Char("Name", size=None, required=True, translate=True)
