diff --git a/tryton/sao/src/common.js b/tryton/sao/src/common.js
index 6eb607bf1f..c43eea570c 100644
--- a/tryton/sao/src/common.js
+++ b/tryton/sao/src/common.js
@@ -798,8 +798,12 @@
                     }
                 });
                 prm.fail(() => {
+                    var selection = [];
+                    if (this.nullable_widget) {
+                        selection.push([null, '']);
+                    }
                     this._last_domain = null;
-                    this.selection = [];
+                    this.selection = selection;
                     if (callback) {
                         callback(this.selection, this.help);
                     }
@@ -845,6 +849,9 @@
         if (!this.attributes.relation) {
             return jQuery.when([]);
         }
+        if (value === null) {
+            return jQuery.when([null, '']);
+        }
         for (var i = 0, len = this.inactive_selection.length; i < len; i++) {
             if (value == this.inactive_selection[i][0]) {
                 return jQuery.when(this.inactive_selection[i]);
