diff --git a/sao/src/tab.js b/sao/src/tab.js
index 24f32bdc..bcf7ac9b 100644
--- a/sao/src/tab.js
+++ b/sao/src/tab.js
@@ -693,13 +693,14 @@
                         menu.children().length);
                 }

-                if (menu_action[0] == 'print') {
-                    if (toolbars.exports.length && toolbars.print.length) {
+                if ((menu_action[0] == 'print') &&
+                    toolbars.exports.length) {
+                    button._can_be_sensitive = true;
+                    if (toolbars.print.length) {
                         menu.append(jQuery('<li/>', {
                             'role': 'separator',
                             'class': 'divider',
                         }));
-                        button._can_be_sensitive = true;
                     }
                     toolbars.exports.forEach(function(export_) {
                         var item = jQuery('<li/>', {
