diff --git a/tryton/modules/account_credit_limit/party.py b/tryton/modules/account_credit_limit/party.py
index ebadd2a3df..3030eeadbf 100644
--- a/tryton/modules/account_credit_limit/party.py
+++ b/tryton/modules/account_credit_limit/party.py
@@ -64,6 +64,11 @@ class Party(metaclass=PoolMeta):
         if self.credit_limit_amount is None:
             return

+        company_id = Transaction().context.get('company')
+        if company_id is None or company_id < 0:
+            return
+        company = Company(company_id)
+
         def in_group():
             group = Group(ModelData.get_id('account_credit_limit',
                     'group_credit_limit'))
@@ -80,7 +85,6 @@ class Party(metaclass=PoolMeta):
             Model = pool.get(model)
             Model.lock()
         if self.credit_limit_amount < self.credit_amount + amount:
-            company = Company(Transaction().context.get('company'))
             lang = Lang.get()
             limit = lang.currency(self.credit_limit_amount, company.currency)
             if not in_group():
@@ -123,7 +127,7 @@ class Party(metaclass=PoolMeta):
         pool = Pool()
         Company = pool.get('company.company')
         company_id = Transaction().context.get('company')
-        if company_id:
+        if company_id is not None and company_id >= 0:
             company = Company(company_id)
             return company.currency.digits
