diff --git a/tryton/modules/stock_product_location/location.py b/tryton/modules/stock_product_location/location.py
index e837912ba3..645ecf7108 100644
--- a/tryton/modules/stock_product_location/location.py
+++ b/tryton/modules/stock_product_location/location.py
@@ -57,7 +57,7 @@ class Move(metaclass=PoolMeta):

     def set_product_location(self, field='to_location', **pattern):
         assert field in {'from_location', 'to_location'}
-        if getattr(self, 'shipment', None):
+        if getattr(self, 'shipment', None) and self.shipment.warehouse:
             pattern.setdefault('warehouse', self.shipment.warehouse.id)
         elif getattr(self, 'production', None):
             pattern.setdefault('warehouse', self.production.warehouse.id)
@@ -72,6 +72,13 @@ class Move(metaclass=PoolMeta):
                 setattr(self, field, product_location.location)
                 break

+    def pick_product(self, quantities):
+        if self.product.consumable and self.warehouse:
+            self.set_product_location(field='from_location',
+                    warehouse=self.warehouse.id)
+        to_pick = super().pick_product(quantities)
+        return to_pick
+

 class ShipmentIn(metaclass=PoolMeta):
     __name__ = 'stock.shipment.in'
