diff --git a/tryton/modules/commission/commission.py b/tryton/modules/commission/commission.py
index 11d04e6863..ac2d629a3a 100644
--- a/tryton/modules/commission/commission.py
+++ b/tryton/modules/commission/commission.py
@@ -261,7 +261,9 @@ class Plan(ModelSQL, ModelView):
         context = self.get_context_formula(amount, product)
         for line in self.lines:
             if line.match(pattern):
-                return line.get_amount(**context)
+                unit_price = line.get_amount(**context)
+                if unit_price is not None:
+                    return round_price(unit_price)


 class PlanLines(sequence_ordered(), ModelSQL, ModelView, MatchMixin):
