diff --git a/tryton/modules/stock_product_location/location.py b/tryton/modules/stock_product_location/location.py
index 453e6a6e5b..e6b4d378c6 100644
--- a/tryton/modules/stock_product_location/location.py
+++ b/tryton/modules/stock_product_location/location.py
@@ -73,6 +73,13 @@ class Move(metaclass=PoolMeta):
                 setattr(self, field, product_location.location)
                 break

+    def pick_product(self, quantities):
+        if self.product.consumable and self.warehouse:
+            self.set_product_location(field='from_location',
+                    warehouse=self.warehouse.id)
+        to_pick = super().pick_product(quantities)
+        return to_pick
+

 class ShipmentIn(metaclass=PoolMeta):
     __name__ = 'stock.shipment.in'
