diff --git a/sao/src/view/form.js b/sao/src/view/form.js
index 032dcb717c..ef6684e191 100644
--- a/sao/src/view/form.js
+++ b/sao/src/view/form.js
@@ -2772,6 +2772,7 @@ function eval_pyson(value){
                     params.name = this.attributes.string;
                     params.context = this.field.get_context(this.record);
                     Sao.Tab.create(params);
+                    this._popup = false;
                     return;
                 }
                 var screen = this.get_screen();
