diff --git a/tryton/sao/src/sao.js b/tryton/sao/src/sao.js
index a9eac0ad99..1fbc27b00c 100644
--- a/tryton/sao/src/sao.js
+++ b/tryton/sao/src/sao.js
@@ -675,6 +675,7 @@ var Sao = {
                                 "Incompatible version of the server."),
                             Sao.i18n.gettext("Version mismatch"));
                     } else {
+                        let url = window.location.hash.substr(1);
                         Sao.Session.get_credentials()
                             .then(function(session) {
                                 Sao.Session.current_session = session;
@@ -684,7 +685,7 @@ var Sao = {
                             .then(function(preferences) {
                                 Sao.menu(preferences);
                                 Sao.user_menu(preferences);
-                                Sao.open_url();
+                                Sao.open_url(url);
                                 Sao.Bus.listen();
                             });
                     }
