diff --git a/tryton/modules/account/tax.py b/tryton/modules/account/tax.py
index 370c456af1..3a7b585954 100644
--- a/tryton/modules/account/tax.py
+++ b/tryton/modules/account/tax.py
@@ -541,7 +541,7 @@ class TaxCodeContext(ModelView):
         return Transaction().context.get('company')

     @fields.depends(
-        'company', 'fiscalyear', 'period', methods=['on_change_with_periods'])
+        'company', 'fiscalyear', 'period', 'periods', methods=['on_change_with_periods'])
     def on_change_company(self):
         if self.fiscalyear and self.fiscalyear.company != self.company:
             self.fiscalyear = None
