diff --git a/tryton/sao/src/common.js b/tryton/sao/src/common.js
index c79eda9aa1..c93586e68f 100644
--- a/tryton/sao/src/common.js
+++ b/tryton/sao/src/common.js
@@ -1198,7 +1198,7 @@
         },
         stringable: function(domain) {
             const stringable_ = clause => {
-                if (!clause) {
+                if (!clause || jQuery.isEmptyObject(clause)) {
                     return true;
                 }
                 var is_array = function(e) {
