diff --git a/tryton/modules/sale_opportunity/opportunity.py b/tryton/modules/sale_opportunity/opportunity.py
index 982810a42d..43ec0abf74 100644
--- a/tryton/modules/sale_opportunity/opportunity.py
+++ b/tryton/modules/sale_opportunity/opportunity.py
@@ -552,6 +552,10 @@ class SaleOpportunityLine(sequence_ordered(), ModelSQL, ModelView):
                 ('salable', '=', True),
                 ()),
             ],
+        context={
+            'company': Eval('company', None),
+            },
+        depends={'company'},
         states=_states)
     product_uom_category = fields.Function(
         fields.Many2One(
@@ -573,6 +577,9 @@ class SaleOpportunityLine(sequence_ordered(), ModelSQL, ModelView):
     description = fields.Text("Description", states=_states)
     summary = fields.Function(fields.Char("Summary"), 'on_change_with_summary')
     note = fields.Text("Note")
+    company = fields.Function(
+        fields.Many2One('company.company', "Company"),
+        'on_change_with_company')

     del _states

@@ -600,6 +607,10 @@ class SaleOpportunityLine(sequence_ordered(), ModelSQL, ModelView):
         if self.opportunity:
             return self.opportunity.state

+    @fields.depends('opportunity', '_parent_opportunity.company')
+    def on_change_with_company(self, name=None):
+        return self.opportunity.company if self.opportunity else None
+
     @fields.depends('product', 'unit')
     def on_change_product(self):
         if not self.product:
