diff --git a/tryton/modules/stock/shipment.py b/tryton/modules/stock/shipment.py
index ac70032b49f5c8ec704c74e12bf435fdff7440ae_bW9kdWxlcy9zdG9jay9zaGlwbWVudC5weQ==..a9b459e6827664b5fdd8027ef093b30555b81245_bW9kdWxlcy9zdG9jay9zaGlwbWVudC5weQ== 100644
--- a/tryton/modules/stock/shipment.py
+++ b/tryton/modules/stock/shipment.py
@@ -146,7 +146,6 @@
     def copy(cls, shipments, default=None):
         default = default.copy() if default is not None else {}
         default.setdefault('number')
-        default.setdefault('moves.origin', None)
         return super().copy(shipments, default=default)
 
 
@@ -642,6 +641,7 @@
             default = {}
         else:
             default = default.copy()
+        default.setdefault('moves.origin', None)
         default.setdefault('received_by', None)
         default.setdefault('done_by', None)
         return super(ShipmentIn, cls).copy(shipments, default=default)
@@ -1001,6 +1001,7 @@
             default = {}
         else:
             default = default.copy()
+        default.setdefault('moves.origin', None)
         default.setdefault('assigned_by', None)
         default.setdefault('done_by', None)
         return super(ShipmentInReturn, cls).copy(shipments, default=default)
@@ -1780,6 +1781,7 @@
             default = {}
         else:
             default = default.copy()
+        default.setdefault('moves.origin', None)
         default.setdefault('picked_by', None)
         default.setdefault('packed_by', None)
         default.setdefault('done_by', None)
@@ -2167,6 +2169,7 @@
         if default is None:
             default = {}
         default = default.copy()
+        default.setdefault('moves.origin', None)
         default.setdefault('received_by', None)
         default.setdefault('done_by', None)
         return super(ShipmentOutReturn, cls).copy(shipments, default=default)
@@ -2666,6 +2669,7 @@
         else:
             default = default.copy()
         default.setdefault('moves', outgoing_moves)
+        default.setdefault('moves.origin', None)
         default.setdefault('moves.from_location', partial(
                 shipment_field, name='from_location'))
         default.setdefault('moves.to_location', partial(
