diff --git a/tryton/modules/production/stock.py b/tryton/modules/production/stock.py
index 02b8408047..7442701ce2 100644
--- a/tryton/modules/production/stock.py
+++ b/tryton/modules/production/stock.py
@@ -164,6 +164,15 @@ class Move(metaclass=PoolMeta):
             self.effective_date = self.production_output.effective_date
         super().set_effective_date()
 
+    @classmethod
+    def copy(cls, moves, default=None):
+        if default is None:
+            default = {}
+        else:
+            default = default.copy()
+        default.setdefault('production_cost_price_updated', False)
+        return super().copy(moves, default=default)
+
     @classmethod
     def on_modification(cls, mode, moves, field_names=None):
         super().on_modification(mode, moves, field_names=field_names)
