--- a/trytond/trytond/modules/account/period.py
+++ b/trytond/trytond/modules/account/period.py
@@ -313,7 +313,7 @@ class Period(Workflow, ModelSQL, ModelView):
             raise ClosePeriodError(
                 gettext('account.msg_close_period_non_posted_moves',
                     period=unposted_move.period.rec_name,
-                    move=unposted_move.rec_name))
+                    moves=unposted_move.rec_name))
         journal_periods = JournalPeriod.search([
             ('period', 'in', [p.id for p in periods]),
             ])