diff --git a/tryton/modules/account_payment_sepa/payment.py b/tryton/modules/account_payment_sepa/payment.py
index c6d40bbd91..ca417a1636 100644
--- a/tryton/modules/account_payment_sepa/payment.py
+++ b/tryton/modules/account_payment_sepa/payment.py
@@ -683,7 +683,7 @@ class Mandate(Workflow, ModelSQL, ModelView):

     @property
     def is_valid(self):
-        if self.state == 'validated':
+        if self.state == 'validated' and self.account_number.active:
             if self.type == 'one-off':
                 if not self.has_payments:
                     return True
