diff --git a/tryton/sao/src/view/tree.js b/tryton/sao/src/view/tree.js
index b6e96adbb37d194c0ba5637b4fe27b1cdf0b2483_c2FvL3NyYy92aWV3L3RyZWUuanM=..5470a58ca38c48a915144784e7582dd1a8cb75ec_c2FvL3NyYy92aWV3L3RyZWUuanM= 100644
--- a/tryton/sao/src/view/tree.js
+++ b/tryton/sao/src/view/tree.js
@@ -2221,6 +2221,6 @@
                 return;
             }
 
-            body = listener = jQuery(document.body);
+            body = jQuery(document.body);
             if (body.hasClass('modal-open')) {
                 listener = this.tree.el.parents('.modal').last();
@@ -2225,5 +2225,7 @@
             if (body.hasClass('modal-open')) {
                 listener = this.tree.el.parents('.modal').last();
+            } else {
+                listener = this.tree.el.parents('.tab-pane').last();
             }
             const handler = event_ => {
                 if ((event_.currentTarget == body[0]) &&
@@ -2236,7 +2238,7 @@
                     event_.stopPropagation();
                     return;
                 }
-                body.off('click.sao.editabletree');
+                listener.off('click.sao.editabletree');
                 this.tree.edit_row(null);
                 return true;
             };
