diff --git a/tryton/modules/account/tax.py b/tryton/modules/account/tax.py
index a7e8495ead..eebdf46314 100644
--- a/tryton/modules/account/tax.py
+++ b/tryton/modules/account/tax.py
@@ -1264,10 +1264,11 @@ class TaxableMixin(object):
             offset_amount = self.currency.rounding.copy_sign(remainder)

             for tax in cycle(taxes.values()):
-                tax['amount'] -= offset_amount
-                remainder -= offset_amount
-                if abs(remainder) < self.currency.rounding:
-                    break
+                if tax['amount']:
+                    tax['amount'] -= offset_amount
+                    remainder -= offset_amount
+                    if abs(remainder) < self.currency.rounding:
+                        break

     @fields.depends('company', methods=['_get_tax_context', '_round_taxes'])
     def _get_taxes(self):
