diff --git a/tryton/modules/stock/product.py b/tryton/modules/stock/product.py
index a2182d579da3e15783d63d62bfc96bf8475112d1_bW9kdWxlcy9zdG9jay9wcm9kdWN0LnB5..b7ad263fbd16f56cf01d3f922e50245c11d75144_bW9kdWxlcy9zdG9jay9wcm9kdWN0LnB5 100644
--- a/tryton/modules/stock/product.py
+++ b/tryton/modules/stock/product.py
@@ -1317,7 +1317,8 @@
                     product=product.rec_name if product else '',
                     exception=exception)) from exception
         if not isinstance(amount, Decimal):
+            product = self.product or self.template
             raise ProductCostPriceError(
                 gettext('stock.msg_invalid_cost_price_not_number',
                     value=amount,
                     cost_price=self.cost_price,
@@ -1320,8 +1321,8 @@
             raise ProductCostPriceError(
                 gettext('stock.msg_invalid_cost_price_not_number',
                     value=amount,
                     cost_price=self.cost_price,
-                    product=product.rec_name))
+                    product=product.rec_name if product else ''))
         return amount
 
     @classmethod
