diff --git a/trytond/trytond/cache.py b/trytond/trytond/cache.py
index b37494cf45..d6330d88dc 100644
--- a/trytond/trytond/cache.py
+++ b/trytond/trytond/cache.py
@@ -329,8 +329,12 @@ class MemoryCache(BaseCache):
                                     Max(table.timestamp),
                                     where=table.name == name))
                         timestamp, = cursor.fetchone()
-                        inst = cls._instances[name]
-                        inst._clear(dbname, timestamp)
+                        try:
+                            inst = cls._instances[name]
+                        except KeyError:
+                            pass
+                        else:
+                            inst._clear(dbname, timestamp)
                 connection.commit()
             finally:
                 database.put_connection(connection)
@@ -410,8 +414,12 @@ class MemoryCache(BaseCache):
                     elif notification.payload:
                         reset = json.loads(notification.payload)
                         for name in reset:
-                            inst = cls._instances[name]
-                            inst._clear(dbname)
+                            try:
+                                inst = cls._instances[name]
+                            except KeyError:
+                                pass
+                            else:
+                                inst._clear(dbname)
                 cls._clean_last = dt.datetime.now()
                 # Keep connected
                 cursor.execute('SELECT 1')
