diff --git a/tryton/sao/src/common.js b/tryton/sao/src/common.js
index 451fd3d022..104aa99ded 100644
--- a/tryton/sao/src/common.js
+++ b/tryton/sao/src/common.js
@@ -1152,8 +1152,7 @@
             string_prefix = string_prefix || '';
             for (var name in fields) {
                 var field = fields[name];
-                if ((field.searchable || (field.searchable === undefined)) &&
-                    (name !== 'rec_name')) {
+                if (field.searchable || (field.searchable === undefined)) {
                     field = jQuery.extend({}, field);
                     var fullname = prefix ? prefix + '.' + name : name;
                     var string = string_prefix ?
@@ -1208,7 +1207,7 @@
                 }
                 var name = clause[0];
                 var value = clause[2];
-                if (name.endsWith('.rec_name')) {
+                if (name.endsWith('.rec_name') && value) {
                     name = name.slice(0, -9);
                 }
                 if (name in this.fields) {
@@ -1257,11 +1256,11 @@
                 var name = clause[0];
                 var operator = clause[1];
                 var value = clause[2];
-                if (name.endsWith('.rec_name')) {
+                if (name.endsWith('.rec_name') && value) {
                     name = name.slice(0, -9);
                 }
                 if (!(name in this.fields)) {
-                    if (this.is_full_text(value)) {
+                    if ((value !== null) && this.is_full_text(value)) {
                         value = value.slice(1, -1);
                     }
                     return this.quote(value);
@@ -1415,7 +1414,7 @@
                 name = clause[0];
                 operator = clause[1];
                 value = clause[2];
-                if (name.endsWith('.rec_name')) {
+                if (name.endsWith('.rec_name') && value) {
                     name = name.substring(0, name.length - 9);
                 }
             }
@@ -1835,7 +1834,7 @@
                         var split = this.split_target_value(field, value);
                         target = split[0];
                         value = split[1];
-                        if (target) {
+                        if (target && value) {
                             field_name += '.rec_name';
                         }
                     } else if (field.type == 'multiselection') {
@@ -1882,7 +1881,7 @@
                         }
                     }
                     if (['many2one', 'one2many', 'many2many', 'one2one',
-                        'many2many', 'one2one'].includes(field.type)) {
+                        'many2many', 'one2one'].includes(field.type) && value) {
                         field_name += '.rec_name';
                     }
                     if (value instanceof Array) {
@@ -2092,7 +2091,13 @@
                 },
                 'selection': convert_selection,
                 'multiselection': convert_selection,
-                'reference': convert_selection,
+                'reference': function() {
+                    if (value === '') {
+                        return null;
+                    } else {
+                        return convert_selection();
+                    }
+                },
                 'datetime': () => Sao.common.parse_datetime(
                     Sao.common.date_format(context.date_format) + ' ' +
                     this.time_format(field), value),
