diff --git a/tryton/modules/purchase/purchase.py b/tryton/modules/purchase/purchase.py
index f5b42c4c76..403c57327b 100644
--- a/tryton/modules/purchase/purchase.py
+++ b/tryton/modules/purchase/purchase.py
@@ -1867,9 +1867,12 @@ class Line(sequence_ordered(), ModelSQL, ModelView):
             if invoice_line.type != 'line':
                 continue
             if invoice_line not in skips:
-                quantity += Uom.compute_qty(
-                    invoice_line.unit or self.unit, invoice_line.quantity,
-                    self.unit)
+                if self.unit:
+                    quantity += Uom.compute_qty(
+                        invoice_line.unit or self.unit, invoice_line.quantity,
+                        self.unit)
+                else:
+                    quantity += invoice_line.quantity
         return quantity

     def _get_invoice_line_moves(self):
diff --git a/tryton/modules/sale/sale.py b/tryton/modules/sale/sale.py
index e25f77bb53..5bb67fdc97 100644
--- a/tryton/modules/sale/sale.py
+++ b/tryton/modules/sale/sale.py
@@ -1903,9 +1903,12 @@ class SaleLine(TaxableMixin, sequence_ordered(), ModelSQL, ModelView):
             if invoice_line.type != 'line':
                 continue
             if invoice_line not in skips:
-                quantity += Uom.compute_qty(
-                    invoice_line.unit or self.unit, invoice_line.quantity,
-                    self.unit)
+                if self.unit:
+                    quantity += Uom.compute_qty(
+                        invoice_line.unit or self.unit, invoice_line.quantity,
+                        self.unit)
+                else:
+                    quantity += invoice_line.quantity
         return quantity

     def _get_invoice_line_moves(self):
