diff --git a/tryton/trytond/trytond/ir/action.py b/tryton/trytond/trytond/ir/action.py
index 230ec10991..16c4860ec6 100644
--- a/tryton/trytond/trytond/ir/action.py
+++ b/tryton/trytond/trytond/ir/action.py
@@ -54,7 +54,8 @@ class Action(DeactivableMixin, ModelSQL, ModelView):
     __name__ = 'ir.action'
     name = fields.Char('Name', required=True, translate=True)
     type = fields.Selection(
-        ACTION_SELECTION, "Type", required=True, readonly=True)
+        ACTION_SELECTION, "Type", required=True,
+        states={'editable': False})
     action = fields.Function(
         fields.Reference("Action", selection=ACTION_SELECTION),
         'get_action')
@@ -345,7 +346,7 @@ class ActionMixin(ModelSQL):
 
     action = fields.Many2One(
         'ir.action', "Action",
-        required=True, readonly=True, ondelete='CASCADE')
+        required=True, states={'editable': False}, ondelete='CASCADE')
 
     @classmethod
     def __setup__(cls):
