diff --git a/tryton/modules/account/move.py b/tryton/modules/account/move.py
index ddbca0ae19..ee5839fcb6 100644
--- a/tryton/modules/account/move.py
+++ b/tryton/modules/account/move.py
@@ -1950,7 +1950,7 @@ class ReconcileLines(Wizard):

     def get_writeoff(self):
         "Return writeoff amount and company"
-        company = None
+        company = currency = None
         amount = Decimal('0.0')
         amount_second_currency = Decimal('0.0')
         second_currencies = set()
@@ -1968,8 +1968,6 @@ class ReconcileLines(Wizard):
         if second_currency:
             amount = amount_second_currency
             currency = second_currency
-        else:
-            currency = company.currency
         return amount, currency, company

     def transition_start(self):
