diff --git a/tryton/modules/account/move.py b/tryton/modules/account/move.py
index ddbca0ae19..aaf5fbfe62 100644
--- a/tryton/modules/account/move.py
+++ b/tryton/modules/account/move.py
@@ -1950,7 +1950,7 @@ class ReconcileLines(Wizard):

     def get_writeoff(self):
         "Return writeoff amount and company"
-        company = None
+        company = currency = None
         amount = Decimal('0.0')
         amount_second_currency = Decimal('0.0')
         second_currencies = set()
@@ -1961,6 +1961,7 @@ class ReconcileLines(Wizard):
             second_currencies.add(line.second_currency)
             if not company:
                 company = line.account.company
+                currency = company.currency
         try:
             second_currency, = second_currencies
         except ValueError:
@@ -1968,8 +1969,6 @@ class ReconcileLines(Wizard):
         if second_currency:
             amount = amount_second_currency
             currency = second_currency
-        else:
-            currency = company.currency
         return amount, currency, company

     def transition_start(self):
