diff --git a/tryton/modules/product/product.py b/tryton/modules/product/product.py
index e46f608dcb..13d04d0212 100644
--- a/tryton/modules/product/product.py
+++ b/tryton/modules/product/product.py
@@ -806,7 +806,7 @@ class ProductCostPrice(ModelSQL, CompanyValueMixin):
         # Migration from 4.4: replace template by product
         if table.column_exist('template'):
             columns = ['create_uid', 'create_date',
-                'write_uid', 'write_date', 'cost_price']
+                'write_uid', 'write_date', 'cost_price', 'company']
             cursor.execute(*sql_table.insert(
                     columns=[Column(sql_table, c) for c in columns]
                     + [sql_table.product],
