diff --git a/tryton/trytond/trytond/backend/postgresql/database.py b/tryton/trytond/trytond/backend/postgresql/database.py
--- a/tryton/trytond/trytond/backend/postgresql/database.py
+++ b/tryton/trytond/trytond/backend/postgresql/database.py
@@ -293,6 +293,13 @@ def get_connection(
                     else:
                         # Detect disconnection
                         cur.execute('SELECT 1')
+                    from pgvector.psycopg2 import register_vector
+                    from psycopg2 import ProgrammingError
+                    try:
+                        register_vector(conn)
+                    except  ProgrammingError as e:
+                        logger.warning('Failed to register pgvector type, '
+                            'pgvector extension may not be installed: "%s"', e)
             except DatabaseOperationalError:
                 self._connpool.putconn(conn, close=True)
                 continue
