diff --git a/product.py b/product.py
index 08537ed..34ccc24 100644
--- a/tryton/modules/sale_supply/product.py
+++ b/tryton/modules/sale_supply/product.py
@@ -45,6 +56,7 @@ class Template(metaclass=PoolMeta):
 
         super().__register__(module)
 
+        '''
         # Migration from 6.6: convert supply_on_sale from boolean to selection
         if migrate_supply_on_sale:
             cursor.execute(*table.update(
@@ -58,6 +70,7 @@ class Template(metaclass=PoolMeta):
                     ['always'],
                     where=h_table._temp_supply_on_sale == Literal(True)))
             h_table_h.drop_column('_temp_supply_on_sale')
+        '''
 
     @fields.depends(methods=['_notify_order_point'])
     def on_change_notify(self):
