diff --git a/tryton/sao/src/sao.js b/tryton/sao/src/sao.js
index 9e5807824e..21bb43ccfa 100644
--- a/tryton/sao/src/sao.js
+++ b/tryton/sao/src/sao.js
@@ -1120,10 +1120,6 @@ var Sao = {
                 shortcut: 'ctrl+p',
                 label: Sao.i18n.gettext('Print'),
                 id: 'print',
-            }, {
-                shortcut: 'ctrl+shift+e',
-                label: Sao.i18n.gettext('E-Mail'),
-                id: 'email',
             }, {
                 shortcut: 'alt+shift+tab',
                 label: Sao.i18n.gettext('Previous tab'),
diff --git a/tryton/sao/src/tab.js b/tryton/sao/src/tab.js
index 411b61a9b6..a47e21f586 100644
--- a/tryton/sao/src/tab.js
+++ b/tryton/sao/src/tab.js
@@ -91,11 +91,6 @@
                     id: 'print',
                     icon: 'tryton-print',
                     label: Sao.i18n.gettext('Print'),
-                }, {
-                    id: 'email',
-                    icon: 'tryton-email',
-                    label: Sao.i18n.gettext('E-Mail...'),
-                    tooltip: Sao.i18n.gettext('Send an e-mail using the record'),
                 }, null, {
                     id: 'export',
                     icon: 'tryton-export',
@@ -532,7 +527,7 @@
                     Sao.i18n.gettext('Open related records')],
                 ['print', 'tryton-print',
                     Sao.i18n.gettext('Print report')]
-            ].forEach(menu_action => {
+            ].reverse().forEach(menu_action => {
                 var dropdown = jQuery('<div/>', {
                     'class': 'btn-group dropdown',
                     'role': 'group'
@@ -558,7 +553,7 @@
                         'role': 'menu',
                         'aria-labelledby': menu_action[0]
                     }))
-                    .insertBefore(toolbar.find('button#email'));
+                    .insertAfter(toolbar.find('button#note'));
                 var button = dropdown.find('button');
                 this.buttons[menu_action[0]] = button;
                 dropdown
@@ -1442,7 +1437,7 @@
             }
             const view_type = this.screen.current_view.view_type;
             const has_views = this.screen.number_of_views > 1;
-            var buttons = ['print', 'relate', 'email', 'attach'];
+            var buttons = ['print', 'relate', 'attach'];
             for (const button_id of buttons) {
                 const button = this.buttons[button_id];
                 let can_be_sensitive = button._can_be_sensitive;
