diff --git a/tryton/sao/src/sao.js b/tryton/sao/src/sao.js
index c33cd423ad..080c0615fb 100644
--- a/tryton/sao/src/sao.js
+++ b/tryton/sao/src/sao.js
@@ -1165,10 +1165,6 @@ var Sao = {
                 shortcut: 'ctrl+p',
                 label: Sao.i18n.gettext('Print'),
                 id: 'print',
-            }, {
-                shortcut: 'ctrl+shift+e',
-                label: Sao.i18n.gettext('Email'),
-                id: 'email',
             }, {
                 shortcut: 'alt+shift+tab',
                 label: Sao.i18n.gettext('Previous tab'),
diff --git a/tryton/sao/src/tab.js b/tryton/sao/src/tab.js
index 4739d0da70..67ea7298f7 100644
--- a/tryton/sao/src/tab.js
+++ b/tryton/sao/src/tab.js
@@ -103,11 +103,6 @@
                     id: 'print',
                     icon: 'tryton-print',
                     label: Sao.i18n.gettext('Print'),
-                }, {
-                    id: 'email',
-                    icon: 'tryton-email',
-                    label: Sao.i18n.gettext('Email...'),
-                    tooltip: Sao.i18n.gettext('Send an email using the record'),
                 }, null, {
                     id: 'export',
                     icon: 'tryton-export',
@@ -591,7 +586,7 @@
                     Sao.i18n.gettext('Open related records')],
                 ['print', 'tryton-print',
                     Sao.i18n.gettext('Print report')]
-            ].forEach(menu_action => {
+            ].reverse().forEach(menu_action => {
                 var dropdown = jQuery('<div/>', {
                     'class': 'btn-group dropdown',
                     'role': 'group'
@@ -617,7 +612,7 @@
                         'role': 'menu',
                         'aria-labelledby': menu_action[0]
                     }))
-                    .insertBefore(toolbar.find('button#email'));
+                    .insertAfter(toolbar.find('button#note'));
                 var button = dropdown.find('button');
                 this.buttons[menu_action[0]] = button;
                 dropdown
@@ -1534,7 +1529,7 @@
             let sensitive = (
                 record_id >= 0 ? record_id !== null && record_id !== undefined :
                 false);
-            var buttons = ['print', 'relate', 'email', 'attach', 'chat'];
+            var buttons = ['print', 'relate', 'attach', 'chat'];
             for (const button_id of buttons) {
                 const button = this.buttons[button_id];
                 if (!button) {
