diff --git a/sao/src/sao.js b/sao/src/sao.js
index 242fa32..cb817dc 100644
--- a/sao/src/sao.js
+++ b/sao/src/sao.js
@@ -1050,10 +1050,6 @@ var Sao = {};
                 shortcut: 'ctrl+p',
                 label: Sao.i18n.gettext('Print'),
                 id: 'print',
-            }, {
-                shortcut: 'ctrl+shift+e',
-                label: Sao.i18n.gettext('E-Mail'),
-                id: 'email',
             }, {
                 shortcut: 'alt+shift+tab',
                 label: Sao.i18n.gettext('Previous tab'),
diff --git a/sao/src/tab.js b/sao/src/tab.js
index d1010ca..7b65bbd 100644
--- a/sao/src/tab.js
+++ b/sao/src/tab.js
@@ -91,11 +91,6 @@
                     id: 'print',
                     icon: 'tryton-print',
                     label: Sao.i18n.gettext('Print'),
-                }, {
-                    id: 'email',
-                    icon: 'tryton-email',
-                    label: Sao.i18n.gettext('E-Mail...'),
-                    tooltip: Sao.i18n.gettext('Send an e-mail using the record'),
                 }, null, {
                     id: 'export',
                     icon: 'tryton-export',
@@ -522,7 +517,7 @@
                     Sao.i18n.gettext('Open related records')],
                 ['print', 'tryton-print',
                     Sao.i18n.gettext('Print report')]
-            ].forEach(function(menu_action) {
+            ].reverse().forEach(function(menu_action) {
                 var dropdown = jQuery('<div/>', {
                     'class': 'btn-group dropdown',
                     'role': 'group'
@@ -548,7 +543,7 @@
                         'role': 'menu',
                         'aria-labelledby': menu_action[0]
                     }))
-                    .insertBefore(toolbar.find('button#email'));
+                    .insertAfter(toolbar.find('button#note'));
                 var button = dropdown.find('button');
                 this.buttons[menu_action[0]] = button;
                 dropdown
@@ -1453,7 +1448,7 @@
                 if (data[0] !== 0) {
                     name = data[0];
                 }
-                var buttons = ['print', 'relate', 'email', 'save', 'attach'];
+                var buttons = ['print', 'relate', 'save', 'attach'];
                 buttons.forEach(function(button_id){
                     var button = this.buttons[button_id];
                     var can_be_sensitive = button._can_be_sensitive;
