diff --git a/tryton/trytond/bin/trytond b/tryton/trytond/bin/trytond
index 0454f5a82a..7c07ab376d 100755
--- a/tryton/trytond/bin/trytond
+++ b/tryton/trytond/bin/trytond
@@ -13,4 +13,5 @@ if os.path.isdir(DIR):
 
 from trytond.cli import main  # noqa: E402
 
-main()
+if __name__ == '__main__':
+    main()
diff --git a/tryton/trytond/bin/trytond-admin b/tryton/trytond/bin/trytond-admin
index ace81439b7..56c014cb85 100755
--- a/tryton/trytond/bin/trytond-admin
+++ b/tryton/trytond/bin/trytond-admin
@@ -13,4 +13,5 @@ if os.path.isdir(DIR):
 
 from trytond.cli.admin import main  # noqa: E402
 
-main()
+if __name__ == '__main__':
+    main()
diff --git a/tryton/trytond/bin/trytond-console b/tryton/trytond/bin/trytond-console
index 7eed855976..480967714e 100755
--- a/tryton/trytond/bin/trytond-console
+++ b/tryton/trytond/bin/trytond-console
@@ -13,4 +13,5 @@ if os.path.isdir(DIR):
 
 from trytond.cli.console import main  # noqa: E402
 
-main()
+if __name__ == '__main__':
+    main()
diff --git a/tryton/trytond/bin/trytond-cron b/tryton/trytond/bin/trytond-cron
index 3ad371a3c3..d45d818494 100755
--- a/tryton/trytond/bin/trytond-cron
+++ b/tryton/trytond/bin/trytond-cron
@@ -13,4 +13,5 @@ if os.path.isdir(DIR):
 
 from trytond.cli.cron import main  # noqa: E402
 
-main()
+if __name__ == '__main__':
+    main()
diff --git a/tryton/trytond/bin/trytond-stat b/tryton/trytond/bin/trytond-stat
index 560fece4fb..796191ea9d 100755
--- a/tryton/trytond/bin/trytond-stat
+++ b/tryton/trytond/bin/trytond-stat
@@ -14,4 +14,5 @@ if os.path.isdir(DIR):
 
 from trytond.cli.stat import main  # noqa: E402
 
-main()
+if __name__ == '__main__':
+    main()
diff --git a/tryton/trytond/bin/trytond-worker b/tryton/trytond/bin/trytond-worker
index fac0c5913c..8725103212 100755
--- a/tryton/trytond/bin/trytond-worker
+++ b/tryton/trytond/bin/trytond-worker
@@ -12,4 +12,5 @@ if os.path.isdir(DIR):
 
 from trytond.cli.worker import main  # noqa: E402
 
-main()
+if __name__ == '__main__':
+    main()
