diff --git a/account.py b/account.py
index c06ccdd..b1182d3 100644
--- a/tryton/modules/account/account.py
+++ b/tryton/modules/account/account.py
@@ -1917,7 +1917,7 @@ class GeneralLedgerAccountContext(ModelView):
     'General Ledger Account Context'
     __name__ = 'account.general_ledger.account.context'
     fiscalyear = fields.Many2One('account.fiscalyear', 'Fiscal Year',
-        required=True,
+        required=False,
         domain=[
             ('company', '=', Eval('company')),
             ],
@@ -1928,7 +1928,8 @@ class GeneralLedgerAccountContext(ModelView):
             ('start_date', '<=', (Eval('end_period'), 'start_date')),
             ],
         states={
-            'invisible': Eval('from_date', False) | Eval('to_date', False),
+            'invisible': (Eval('from_date', False) | Eval('to_date', False)
+                | ~Eval('fiscalyear', False))
             },
         depends=['fiscalyear', 'end_period', 'from_date', 'to_date'])
     end_period = fields.Many2One('account.period', 'End Period',
@@ -1937,7 +1938,8 @@ class GeneralLedgerAccountContext(ModelView):
             ('start_date', '>=', (Eval('start_period'), 'start_date'))
             ],
         states={
-            'invisible': Eval('from_date', False) | Eval('to_date', False),
+            'invisible': (Eval('from_date', False) | Eval('to_date', False)
+                | ~Eval('fiscalyear', False)),
             },
         depends=['fiscalyear', 'start_period', 'from_date', 'to_date'])
     from_date = fields.Date("From Date",
@@ -2155,7 +2157,7 @@ class GeneralLedgerLine(ModelSQL, ModelView):
     def __setup__(cls):
         super(GeneralLedgerLine, cls).__setup__()
         cls.__access__.add('account')
-        cls._order.insert(0, ('date', 'ASC'))
+        cls._order = [('date', 'DESC'), ('id', 'DESC')]
 
     @classmethod
     def table_query(cls):
