diff --git a/modules/account_invoice/invoice.py b/modules/account_invoice/invoice.py
index 1439992bd9..d4f2a451d2 100644
--- a/tryton/modules/account_invoice/invoice.py
+++ b/tryton/modules/account_invoice/invoice.py
@@ -2505,8 +2505,7 @@ class InvoiceLine(sequence_ordered(), ModelSQL, ModelView, TaxableMixin):
                 else self.currency)
             amount = (Decimal(str(self.quantity or 0))
                 * (self.unit_price or Decimal(0)))
-            invoice_type = (
-                self.invoice.type if self.invoice else self.invoice_type)
+            invoice_type = self.invoice.type if self.invoice else None
             if (invoice_type == 'in'
                     and self.taxes_deductible_rate is not None
                     and self.taxes_deductible_rate != 1):
