diff --git a/tryton/trytond/trytond/model/fields/binary.py b/tryton/trytond/trytond/model/fields/binary.py
index 94919c0b69..370f77808a 100644
--- a/tryton/trytond/trytond/model/fields/binary.py
+++ b/tryton/trytond/trytond/model/fields/binary.py
@@ -1,5 +1,7 @@
 # This file is part of Tryton.  The COPYRIGHT file at the top level of
 # this repository contains the full copyright notices and license terms.
+import logging
+
 from sql import Column, Null
 
 from trytond.filestore import filestore
@@ -8,6 +10,8 @@ from trytond.transaction import Transaction
 
 from .field import Field
 
+logger = logging.getLogger(__name__)
+
 
 def caster(d):
     if isinstance(d, bytes):
@@ -88,7 +92,7 @@ class Binary(Field):
                     try:
                         res[record_id] = store_func(file_id, prefix)
                     except (IOError, OSError):
-                        pass
+                        logger.exception( "failed to retrieve %r from filestore at %r", file_id, prefix)
 
         for i in values:
             if i['id'] in res:
