diff --git a/trytond/worker.py b/trytond/worker.py
index 70f4ba45..99b5a5f7 100644
--- a/trytond/trytond/worker.py
+++ b/trytond/trytond/worker.py
@@ -121,6 +121,9 @@ def run_task(pool, task_id):
                         transaction.rollback()
                         continue
                     raise
+                except Exception:
+                    logger.critical('%s failed', name, exc_info=True)
+                    return
         logger.info('%s done', name)
     except backend.DatabaseOperationalError:
         logger.info('%s failed, retrying', name, exc_info=True)
@@ -136,5 +139,3 @@ def run_task(pool, task_id):
         except Exception:
             logger.critical(
                 'rescheduling %s failed', name, exc_info=True)
-    except Exception:
-        logger.critical('%s failed', name, exc_info=True)
