diff --git a/trytond/trytond/worker.py b/trytond/trytond/worker.py
index b0d751fb..16070d9b 100644
--- a/trytond/trytond/worker.py
+++ b/trytond/trytond/worker.py
@@ -143,6 +143,9 @@ def run_task(pool, task_id):
                 except (UserError, UserWarning) as e:
                     Error.log(task, e)
                     raise
+                except Exception:
+                    logger.critical('%s failed', name, exc_info=True)
+                    return
         logger.info("%s in %i ms", name, duration())
     except backend.DatabaseOperationalError:
         logger.info(
@@ -170,6 +173,3 @@ def run_task(pool, task_id):
         logger.info(
             "%s failed after %i ms", name, duration(),
             exc_info=logger.isEnabledFor(logging.DEBUG))
-    except Exception:
-        logger.critical(
-            "%s failed after %i ms", name, duration(), exc_info=True)
