diff --git a/tryton/trytond/trytond/worker.py b/trytond/trytond/worker.py
index 5ce8ef0d..b5921908 100644
--- a/tryton/trytond/trytond/worker.py
+++ b/tryton/trytond/trytond/worker.py
@@ -139,6 +139,9 @@ def run_task(pool, task_id):
                 except (UserError, UserWarning) as e:
                     Error.log(task, e)
                     raise
+                except Exception:
+                    logger.critical('%s failed', name, exc_info=True)
+                    return
         logger.info('%s done', name)
     except backend.DatabaseOperationalError:
         logger.info('%s failed, retrying', name, exc_info=True)
@@ -162,5 +165,3 @@ def run_task(pool, task_id):
                 'rescheduling %s failed', name, exc_info=True)
     except (UserError, UserWarning):
         logger.info('%s failed', name)
-    except Exception:
-        logger.critical('%s failed', name, exc_info=True)
